---
title: "deseq2_GSE162069"
author: "Ryan Rebernick"
date: "3/29/2021"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


## Description:


## Load packages
```{r, warning=F, message=FALSE}

library('plyr')
library('dplyr') 
library('stringr')
library('data.table')
library('tidyverse')
library('magrittr')

library('DESeq2')
library('tximport')
library('tximportData')
library('rhdf5')
library('EnhancedVolcano')
library('cowplot')
library('ggfortify')

```


## Transcript to gene
Used ensembl v97 - the same as used to create the index file for kallisto
```{r}

library('biomaRt')

# archive ensembl to match the kalliso index 
listEnsemblArchives()
listEnsembl(version = 97)
ensembl97 <- useEnsembl(biomart = 'ensembl', 
                       dataset = 'hsapiens_gene_ensembl',
                       version = 97)

# look at attributes/filters
listAttributes(ensembl97)
searchFilters(mart = ensembl97, pattern = "chromosome")

# get transcripts and genes
tx2gene <- getBM(attributes = c('ensembl_transcript_id_version', 'ensembl_gene_id_version'),
      filters = 'chromosome_name',
      values = c(1:22, 'X', 'Y'), 
      mart = ensembl97,
      useCache = F)

tx2gene <- getBM(attributes = c('ensembl_transcript_id_version', 'external_gene_name'),
      filters = 'chromosome_name',
      values = c(1:22, 'X', 'Y'), 
      mart = ensembl97,
      useCache = F)


```

# All samples
Check to see if is better to include all samples or just ones directly being compared. All samples looked @ first. 

## Import from kallisto and run DESeq2
```{r}

# access the file paths from kallisto output
dir <- '/mctp/share/users/rebernrj/projects/ferroptosisGeneSig/int/kallisto/GSE162069/'
files <- list.files(dir, pattern = 'abundance.h5', recursive = T)
files <- paste0(dir, files)
names(files) <- list.files(dir)

# import at gene level
txi.kallisto <- tximport(files, type = "kallisto", tx2gene = tx2gene)
head(txi.kallisto$counts)

# create sample table for import from tximport
sampleTable <- data.frame(condition = factor(c(rep("MDAMB231", 2), rep("MDAMB231_ML162", 3), rep("MDAMB231_ESA", 3))))
rownames(sampleTable) <- colnames(txi.kallisto$counts)

# import into DESeq2
dds <- DESeqDataSetFromTximport(txi.kallisto, sampleTable, ~condition)

# Run DESeq2
dds <- DESeq(dds)


```

## Differential expression analysis of erastin vs control using only erastin/control samples
```{r}

# get DE
res <- results(dds, contrast=c("condition","MDAMB231_ML162","MDAMB231"))
summary(res)

# save for comparison
hold <- res


```


# Only samples being compared

## Import from kallisto and run DESeq2
```{r}

# access the file paths from kallisto output
dir <- '/mctp/share/users/rebernrj/projects/ferroptosisGeneSig/int/kallisto/GSE162069/'
files <- list.files(dir, pattern = 'abundance.h5', recursive = T)
files <- paste0(dir, files)
names(files) <- list.files(dir)

# select samples being compared
files <- files[1:5]

# import at gene level
txi.kallisto <- tximport(files, type = "kallisto", tx2gene = tx2gene)
head(txi.kallisto$counts)

# create sample table for import from tximport
sampleTable <- data.frame(condition = factor(c(rep("MDAMB231", 2), rep("MDAMB231_ML162", 3))))
rownames(sampleTable) <- colnames(txi.kallisto$counts)

# import into DESeq2
dds <- DESeqDataSetFromTximport(txi.kallisto, sampleTable, ~condition)

# Run DESeq2
dds <- DESeq(dds)


```

## Differential expression analysis of erastin vs control using only erastin/control samples
Better to only include samples in direct comparison
```{r}

# get DE
res <- results(dds, contrast=c("condition","MDAMB231_ML162","MDAMB231"))

summary(res)
summary(hold)

```

## QC plots
```{r}
# distribution of p values
hist(res$pvalue, main = 'GSE162069 p-values')

# Volcano plot
EnhancedVolcano(res,
                lab = rownames(res),
                x = 'log2FoldChange',
                y = 'pvalue')

# MA-plot
## regular lfc
plotMA(res, ylim=c(-5,5), main = 'MA-plot LFCs')

## shrunken lfc
resultsNames(dds)
resLFC <- lfcShrink(dds, coef="condition_MDAMB231_ML162_vs_MDAMB231", type="apeglm")
plotMA(resLFC, ylim=c(-2,2), main = 'MA-plot Shrunken LFCs')

# PCA
dat <- as.data.frame(t(txi.kallisto$counts))
dat$group <- c(rep("MDAMB231", 2), rep("MDAMB231_ML162", 3))
dat$sample <- rownames(dat)
autoplot(dat %>% dplyr::select(-group, -sample) %>% prcomp(),
         data = dat,
         colour = 'group',
         size = 6) +
  theme_cowplot(12)

```

## Combine with count info and save
```{r}

# convert to data frame
res <- as.data.frame(res@listData, row.names = res@rownames)

# combine with count info
res_counts <- merge(res, txi.kallisto$counts, by = 0)
res_counts %<>% arrange(padj)

# save 
write.table(res_counts, '/mctp/share/users/rebernrj/projects/ferroptosisGeneSig/int/GSE162069/deseq2/GSE162069_degs_erastin_control.txt',
            row.names = T, col.names = T, quote = F, sep = '\t')


```


