---
title: "eda_nbl_cohort"
author: "Ryan Rebernick"
date: "4/4/2021"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, fig.keep = 'all')
```


# Description:
This script explores the NBL cohort by using the created ferroptosis gene signatures to explore scores in comparisons of interest. These include: 
- LR vs IR vs HR
- Dinutuximab response (Group 1-4)
- Pre/Post Dinutuximab
- Event-free survival
  - Split by mean
  - Split by quartile
- Overall survival
  - Split by mean
  - Split by quartile
Comparisons were done using 4 gene signatures created for ferroptosis. These gene signatures were scored using 3 methods: Nondirectional Z scores, Directional Z-scores, and SingScore (https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-018-2435-4). 


# Load packages
```{r, warning=F, message=FALSE}

library('plyr')
library('dplyr')
library('stringr')
library('data.table')
library('tidyverse')
library('magrittr')

#BiocManager::install("singscore")
library('singscore')
#install.packages("ggpubr")
library('ggpubr')
library('ggfortify')
library('cowplot')
library(survival)
library(matrixStats)

```


# Load NBL cohort data and select samples for analysis
## Load NBL counts data and merge with metadata
17 duplicate gene names are present. These are just removed. If they are needed for any of the gene signatures they can be added to the dataset later. 
```{r}

# Load RNA-seq counts
nbl <- fread('/mctp/share/users/rebernrj/projects/ferroptosisGeneSig/data/nbl_cohort/release/v1/nbl_expression-tpm.csv', data.table = F)

# Remove any duplicate gene_ids so can transpose matrix
## View duplicate common gene_ids
duplicates <- nbl$gene_name[nbl$gene_name %>% duplicated()]
## remove duplicates
nbl %<>% dplyr::filter(!gene_name %in% duplicates)

# reformat the RNA-seq counts for merging with the clinical data
nbl %<>% set_rownames(nbl$gene_name) %>%
  dplyr::select(-gene_name, -gene_id) %>%
  t() %>%
  as.data.frame()

# Load the clinical data
#temp <- fread('/mctp/share/users/rebernrj/projects/ferroptosisGeneSig/data/nbl_cohort/release/ForRyan/20210403_ForRyan_2021_0403_nbl_rnaseq.txt', data.table = F)
temp <- fread('/mctp/share/users/rebernrj/projects/ferroptosisGeneSig/data/nbl_cohort/release/ForRyan/2021_0416_nbl_ryan.txt', data.table = F)

# Merge dataframes for downstream use
nbl <- merge(temp, nbl, by.x = 'rlib', by.y = 0)


```

## Select samples needed
Filtered for samples based on the following criteria:
- RNA capture method of 'capt'; ensured uniform RNA capture method
- Nonzero purity scores
- For rebiopsies of patient, took the sample with the higher purity w/in a timepoint (i.e. dxprim, dx2prim, ect)
- Eliminated samples which didn't fit the HR/LR/IR criteria (1 sample)
```{r}

# Filter for desired samples
nbl %<>% 
  ## Filter for the 'capt' RNA capture method
  dplyr::filter(r.capture == 'capt') %>%
  ## filter for nonzero purity of RNA
  dplyr::filter(purity > quantile(purity)[3]) %>%
  ## Combine timepoint refrel and relapse variables
  dplyr::mutate(timepoint = ifelse(timepoint %in% c('refrel', 'relapse'), 'rel_ref', timepoint)) %>%
  ## Generalize the timepoint variables
  dplyr::mutate(genTime = ifelse(timepoint %in% c('dxprim', 'dx2prim', 'dxmet'), 'dx', timepoint)) %>%
  ## Arrange by RNA purity for duplicate removal
  dplyr::arrange(timepoint, -purity) %>%
  ## Create pseudo variable for duplicate removal (ex. PO_3133)
  dplyr::mutate(pt.time = paste0(patient, '.', timepoint))
## show duplicates
nbl$patient[duplicated(nbl$pt.time)]
## remove duplicates based on purity, keeping highest purity
nbl <- nbl[!duplicated(nbl$pt.time),]
## Ensure samples represent only risk rated groups
nbl %<>% dplyr::filter(group %in% c('HR', 'IR', 'LR'))

# Create comparison variable for future use
nbl$comp <- ifelse(nbl$group %in% c('IR', "LR"), 'IR/LR', 'HR')

# Order dataframe cols
nbl %<>% dplyr::select(rlib:MYCN.cycle, comp, genTime, ARF5:AC106791.3)


```



# General QC of NBL cohort

### PCA by sample origin
```{r}

# All genes
set.seed(11)
nbl$comp <- nbl$cohort
dat <- nbl %>% dplyr::select(comp, ARF5:AC106791.3)
autoplot(dat %>% dplyr::select(-comp) %>% prcomp(),
              data = dat,
              colour = 'comp',
              main = paste0('PCA of all genes split by cohort')) +
  theme_cowplot(12)

# PCA using top 25% most variable genes
## get most variable features
comp <- nbl$comp
dat <- nbl %>% 
  dplyr::select(ARF5:AC106791.3) %>% 
  t() %>% 
  as.data.frame()
dat %<>% mutate(sd = rowSds(as.matrix(.), na.rm = T)) %>%
  dplyr::filter(sd > quantile(sd)[4]) %>%
  dplyr::select(-sd) %>%
  t() %>%
  as.data.frame() %>%
  cbind(comp, .)
## plot
set.seed(11)
autoplot(dat %>% dplyr::select(-comp) %>% prcomp(),
              data = dat,
              colour = 'comp',
              main = paste0('PCA of 25% most variable genes split by cohort')) +
  theme_cowplot(12)

```



# Create functions neeed to analyze gene signatures

## Create a function to return a signature scores using 3 methods:
1) Sample scoring package - less dependence of score on other samples
https://bioconductor.org/packages/devel/bioc/vignettes/singscore/inst/doc/singscore.html#21_Sample_scoring
2) Nondirectional Summation of Z scores
- Sum the absolute values of Z scores across up and downregulated genes
3) Directional Summation of Z scores
- Sum z scores of upregulated genes + -1*(Sum of Z scores of downregulated genes)
```{r}

# DF - the cleaned nbl dataframe from above
# upGenes - list of genes upregulated in gene signature
# DOWNGENES - list of genes downregulated in gene signature
# METACOLS - the columns of metadata within DF

calculate_signature_scores <- function(DF = nbl, UPGENES, DOWNGENES, METACOLS = 1:33){
  
  # create temp dataframe
  cnts <- DF %>% set_rownames(paste0('Sample_', 1:length(DF$rlib)))
  
  # Score samples using singScore and merge
  ## score
  rankData <- cnts %>% dplyr::select(-METACOLS) %>%  t() %>% rankGenes()
  if(!is.null(UPGENES) & !is.null(DOWNGENES)){
    scoredf <- simpleScore(rankData, upSet = UPGENES, downSet = DOWNGENES)
  } else if(is.null(UPGENES) & !is.null(DOWNGENES)) {
    scoredf <- simpleScore(rankData, downSet = DOWNGENES)
    print('Only DOWN geneset is valid')
  } else if(!is.null(UPGENES) & is.null(DOWNGENES)) {
    scoredf <- simpleScore(rankData, upSet = UPGENES)
    print('Only UP geneset is valid')
  } else {
    print('Both gene sets NULL')
    break
  }
  nms <- paste0('ss.', colnames(scoredf))
  colnames(scoredf) <- nms
  ## merge SingScore Total score with tmp df
  cnts <- merge(cnts, scoredf, by = 0)
  
  
  # Calcluate nondirectional z scores 
  temp <- apply(cnts[,c(UPGENES, DOWNGENES)], 2, scale)
  temp$nondirectional_Zsum <- rowSums(abs(temp))
  ## Add Z score to dataframe 
  cnts$nondirectional_Zsum <- temp$nondirectional_Zsum
  
  
  # Calculate directional Z scores
  ## Upregulated genes
  temp <- apply(cnts %>% dplyr::select(all_of(UPGENES)), 2, scale)
  if(!is_empty(temp)) {
    temp$upZsum <- rowSums(temp)
    ## Add Z score to dataframe 
    cnts$upZsum <- temp$upZsum
  } else {
    cnts$upZsum <- 0
  }
  ## Downregulated genes
  temp <- apply(cnts %>% dplyr::select(all_of(DOWNGENES)), 2, scale)
  if(!is_empty(temp)) {
    temp$downZsum <- rowSums(temp)
    ## Add Z score to dataframe 
    cnts$downZsum <- temp$downZsum
  } else {
    cnts$downZsum <- 0
  }
  
  # Select columns
  cnts %<>% dplyr::select(METACOLS, all_of(nms), upZsum, downZsum)
  cnts %<>% mutate(directional_Zsum = upZsum + (downZsum * -1)) %>%
    dplyr::select(-Row.names)
  
  # Correlation
  print(paste0('Correlation b/t SingScore and Directional Zsum: ', cor(cnts$ss.TotalScore, cnts$directional_Zsum)))
  
  # Return
  return(cnts)
  
  
}

```

## Function to plot comparisons of interest
```{r}

# DF - dataframe to use; nbl format
# UPGENES - list of upregulated genes
# DOWNGENES - list of downregulated genes
# COMP - comparison variable/column
# STAT_TEST - c('wilcox.test', 't.test', 'anova', 'kruskal.test')
# TIMEPOINT = timepoint in nbl cohort to look at
make_plots <- function(
  DF,
  UPGENES, 
  DOWNGENES, 
  COMP='comp', 
  TIMEPOINT, 
  STAT_TEST = 'wilcox.test',
  TITLE = 'XX gene sig'){
  
  ####################
  # return genes not in provided gene list and remove from query
  ####################
  print(paste0('Up genes not in count matrix: ', UPGENES[!UPGENES %in% colnames(DF)]))
  print(paste0('Down genes not in count matrix: ', DOWNGENES[!DOWNGENES %in% colnames(DF)]))
  UPGENES <- UPGENES[UPGENES %in% colnames(DF)]
  DOWNGENES <- DOWNGENES[DOWNGENES %in% colnames(DF)]
  
  ####################
  # PCA PLOT by group
  ####################
  set.seed(11)
  dat <- DF[, c(COMP, UPGENES, DOWNGENES)]
  a <- autoplot(dat %>% dplyr::select(-COMP) %>% prcomp(),
           data = dat,
           colour = COMP,
           main = paste0('PCA using ', TITLE)) +
    theme_cowplot(12)
  
  
  ####################
  # Calclulate scores
  ####################
  scores <- calculate_signature_scores(DF = DF, 
                                   UPGENES = UPGENES,
                                   DOWNGENES = DOWNGENES)
  
  ####################
  # Boxplot 
  ####################

  # select samples
  if(TIMEPOINT %in% c('dx2prim', 'dxmet', 'dxprim', 'posttx', 'rel_ref')){
    DF %<>% dplyr::filter(timepoint %in% c(TIMEPOINT))
  } else if(TIMEPOINT == 'all'){
    print('All timepoints included')
  } else {
    break
  }
  
  # Plot 
  b1 <- scores %>% ggplot(aes(x=get(COMP), y=ss.TotalScore)) + 
    geom_boxplot(outlier.shape = NA) +
    geom_jitter(alpha=0.6, width=0.15) +
    theme_bw(base_size=16) +
    labs(subtitle=TITLE) +
    stat_compare_means(method = STAT_TEST) +
    xlab(NULL) +
    theme(axis.text.x=element_text(angle = -90, hjust = 0))
  
  # Plot 
  # b2 <- scores %>% ggplot(aes(x=get(COMP), y=nondirectional_Zsum)) + 
  #   geom_boxplot() +
  #   geom_jitter(alpha=0.6, width=0.15) +
  #   theme_bw(base_size=16) +
  #   labs(subtitle="Nondirectional Z score by Risk group in dxprim") +
  #   stat_compare_means(method = STAT_TEST) +
  #   xlab(NULL)
  
  # Plot 
  # b3 <- scores %>% ggplot(aes(x=get(COMP), y=directional_Zsum)) +
  #   geom_boxplot(outlier.shape = NA) +
  #   geom_jitter(alpha=0.6, width=0.15) +
  #   theme_bw(base_size=16) +
  #   labs(subtitle=paste0("Directional Z score by Risk group in dxprim: ", TITLE)) +
  #   stat_compare_means(method = STAT_TEST) +
  #   xlab(NULL)
  
  return(list(scores,a,b1))
  
}

```



# Analyze gene signatures 

## 306 gene signature

### Load gene signature
```{r}
sig306 <- fread('/mctp/share/users/rebernrj/projects/ferroptosisGeneSig/out/gene_signatures/erastin_clue24h_gse104462_306genes.txt',
                data.table = F)

# Upregulated
sig306_up <- sig306 %>% 
  dplyr::filter(upregulated == T) %>% 
  dplyr::select(V1) %>% unlist()

# Downregulated
sig306_dn <- sig306 %>% 
  dplyr::filter(upregulated == F) %>% 
  dplyr::select(V1) %>% unlist()


```

### Distribution of 306 gene signature scores in all patients. Correlation with SLC levels
```{r}

# Define comparison
nbl$comp <- 'all'

# get PCA and boxplots
cur <- make_plots(
  DF = nbl,
  UPGENES = sig306_up, 
  DOWNGENES = sig306_dn, 
  COMP = 'comp', 
  TIMEPOINT = 'all', 
  STAT_TEST = 'anova',
  TITLE = '306 gene sig')

# Plot distribution of sing score
cur[3]
#cur[4]

# Look at correlation of singscore with SLC levels
cur <- cur[[1]]
cur %<>% dplyr::select(rlib, ss.TotalScore, directional_Zsum) %>%
  merge(., nbl, by = 'rlib')

# 306gs singScore with SLC3A2 expression
ggscatter(cur, x='ss.TotalScore', y='SLC3A2',
          add='reg.line',
          add.params = list(color='blue', fill='lightgray'),
          conf.int = T) +
  stat_cor(method = "pearson")

# 306gs singScore with SLC7A11
ggscatter(cur, x='ss.TotalScore', y='SLC7A11',
          add='reg.line',
          add.params = list(color='blue', fill='lightgray'),
          conf.int = T) +
  stat_cor(method = "pearson")

# Directional Z sum with SLC3A2
# ggscatter(cur, x='directional_Zsum', y='SLC3A2',
#           add='reg.line',
#           add.params = list(color='blue', fill='lightgray'),
#           conf.int = T) +
#   stat_cor(method = "pearson")

# Directional Z sum with SLC7A11
# ggscatter(cur, x='directional_Zsum', y='SLC7A11',
#           add='reg.line',
#           add.params = list(color='blue', fill='lightgray'),
#           conf.int = T) +
#   stat_cor(method = "pearson")

# SLC7A11 vs SLC3A2
ggscatter(cur, x='SLC3A2', y='SLC7A11',
          add='reg.line',
          add.params = list(color='blue', fill='lightgray'),
          conf.int = T) +
  stat_cor(method = "pearson")

```

### Correlation with NK cell infiltration
NK gene signature taken from https://cancerimmunolres.aacrjournals.org/content/7/7/1162. Paper by same authors of singscore. 
```{r}

# Define comparison
nbl$comp <- 'all'

# get NK cell gene signature
nk <- fread('/mctp/share/users/rebernrj/projects/ferroptosisGeneSig/data/doi_10.1158_2326-6966.CIR-18-0500_NKcellGeneSig/s1.txt', data.table = F, skip = 2)
nk %<>% dplyr::select(`HGNC Symbol`, `Cursons\nGuimaraes\nsigGene`, logFC) %>%
  set_colnames(c('gene', 'sig', 'logfc'))
## upregulated
nk_up <- nk %>% dplyr::filter(logfc > 0) %>%
  dplyr::filter(sig == T) %>%
  dplyr::select(gene) %>%
  unlist()

# Obtain nk cell gene signature scores across NBL samples
cur <- make_plots(
  DF = nbl,
  UPGENES = nk_up, 
  DOWNGENES = NULL, 
  COMP = 'comp', 
  TIMEPOINT = 'all', 
  STAT_TEST = 'anova',
  TITLE = 'NK gene sig')
cur[[3]]
nk.ss <- cur[[1]]
nk.ss.sub <- nk.ss %>% dplyr::select(rlib, ss.TotalScore, directional_Zsum) %>%
  set_colnames(c('rlib', 'nk.ss.TotalScore', 'nk.directional_Zsum'))

# get sig306 gene sig
cur <- make_plots(
  DF = nbl,
  UPGENES = sig306_up, 
  DOWNGENES = sig306_dn, 
  COMP = 'comp', 
  TIMEPOINT = 'all', 
  STAT_TEST = 'anova',
  TITLE = '306 gene sig')

# Look at correlation of singscore with NK score levels
cur <- cur[[1]]
cur %<>% dplyr::select(rlib, ss.TotalScore, directional_Zsum) %>%
  merge(., nk.ss.sub, by = 'rlib')
ggscatter(cur, x='ss.TotalScore', y='nk.ss.TotalScore',
          add='reg.line',
          add.params = list(color='blue', fill='lightgray'),
          conf.int = T) +
  stat_cor(method = "pearson")

# View NK score vs Dinutux ever
nk.ss %<>% mutate(comp = ifelse(dinutx_ever ==1, 'yes', 'no'))
nk.ss %>% ggplot(aes(x=comp, y=ss.TotalScore)) + 
    geom_boxplot(outlier.shape = NA) +
    geom_jitter(alpha=0.6, width=0.15) +
    theme_bw(base_size=16) +
    labs(subtitle='NK score vs Dinutux ever') +
    stat_compare_means(method = 'wilcox.test') +
    xlab(NULL)

```

### Correlation with cell proportions and Gini coefficients
```{r}

# Load T cell data
epic <- fread('/mctp/share/users/rebernrj/projects/ferroptosisGeneSig/out/EPIC/NBL/cell_proportions/nbl_epic_proportions_all.txt', data.table = F) %>%
  dplyr::select(rlib, Bcells:otherCells)
tcr <- fread('/mctp/share/users/rebernrj/projects/ferroptosisGeneSig/data/nbl_cohort/other/nbl-crisp-canio_tcr-summary.csv', data.table = F)


# Load gene sig scores
cur <- make_plots(
  DF = nbl,
  UPGENES = sig306_up, 
  DOWNGENES = sig306_dn, 
  COMP = 'comp', 
  TIMEPOINT = 'all', 
  STAT_TEST = 'anova',
  TITLE = '306 gene sig')
cur <- cur[[1]]

# Combine T cell data with gene signature 
comb <- unique(merge(cur, epic, by = 'rlib', all.x = T))
comb <- merge(comb, tcr, by.x = 'rlib', by.y='sid', all.x=T)

# T cell clones
ggscatter(comb, x='ss.TotalScore', y='clones_T',
          add='reg.line',
          add.params = list(color='blue', fill='lightgray'),
          conf.int = T) +
  stat_cor(method = "pearson")

# B cell clones
ggscatter(comb, x='ss.TotalScore', y='clones_B',
          add='reg.line',
          add.params = list(color='blue', fill='lightgray'),
          conf.int = T) +
  stat_cor(method = "pearson")

# T cell Gini
ggscatter(comb, x='ss.TotalScore', y='gini_T',
          add='reg.line',
          add.params = list(color='blue', fill='lightgray'),
          conf.int = T) +
  stat_cor(method = "pearson")

# B cell Gini
ggscatter(comb, x='ss.TotalScore', y='gini_B',
          add='reg.line',
          add.params = list(color='blue', fill='lightgray'),
          conf.int = T) +
  stat_cor(method = "pearson")

# CD4
ggscatter(comb, x='ss.TotalScore', y='CD4_Tcells',
          add='reg.line',
          add.params = list(color='blue', fill='lightgray'),
          conf.int = T) +
  stat_cor(method = "pearson")

# CD8
ggscatter(comb, x='ss.TotalScore', y='CD8_Tcells',
          add='reg.line',
          add.params = list(color='blue', fill='lightgray'),
          conf.int = T) +
  stat_cor(method = "pearson")

# B cells
ggscatter(comb, x='ss.TotalScore', y='Bcells',
          add='reg.line',
          add.params = list(color='blue', fill='lightgray'),
          conf.int = T) +
  stat_cor(method = "pearson")

# CAFs
ggscatter(comb, x='ss.TotalScore', y='CAFs',
          add='reg.line',
          add.params = list(color='blue', fill='lightgray'),
          conf.int = T) +
  stat_cor(method = "pearson")

# Endothelial
ggscatter(comb, x='ss.TotalScore', y='Endothelial',
          add='reg.line',
          add.params = list(color='blue', fill='lightgray'),
          conf.int = T) +
  stat_cor(method = "pearson")

# Macrophages
ggscatter(comb, x='ss.TotalScore', y='Macrophages',
          add='reg.line',
          add.params = list(color='blue', fill='lightgray'),
          conf.int = T) +
  stat_cor(method = "pearson")

# NKcells
ggscatter(comb, x='ss.TotalScore', y='NKcells',
          add='reg.line',
          add.params = list(color='blue', fill='lightgray'),
          conf.int = T) +
  stat_cor(method = "pearson")

# otherCells
ggscatter(comb, x='ss.TotalScore', y='otherCells',
          add='reg.line',
          add.params = list(color='blue', fill='lightgray'),
          conf.int = T) +
  stat_cor(method = "pearson")


```

### Look for factors common for the highest quartile ferroptosis scores
```{r}

# get df of scores
cur <- make_plots(
  DF = nbl,
  UPGENES = sig306_up, 
  DOWNGENES = sig306_dn, 
  COMP = 'comp', 
  TIMEPOINT = 'dxprim', 
  STAT_TEST = 'anova',
  TITLE = '306gs: ')

dat <- cur[[1]] %>%
  mutate(ss.quartile = ifelse(ss.TotalScore >= quantile(ss.TotalScore)[3], 3, 2)) %>%
  mutate(ss.quartile = ifelse(ss.TotalScore >= quantile(ss.TotalScore)[4], 4, ss.quartile)) %>%
  mutate(ss.quartile = ifelse(ss.TotalScore < quantile(ss.TotalScore)[2], 1, ss.quartile)) 

#View(dat %>% dplyr::filter(ss.quartile == 4))


```

### 306 gene signature for LR vs IR vs HR
```{r}

# Define comparison
nbl$comp <- nbl$group

# get PCA and boxplots
cur <- make_plots(
  DF = nbl,
  UPGENES = sig306_up, 
  DOWNGENES = sig306_dn, 
  COMP = 'comp', 
  TIMEPOINT = 'dxprim', 
  STAT_TEST = 'anova',
  TITLE = '306gs: LR/IR/HR')

# plot and get df
cur[2]
cur[3]
#cur[4]
# cur <- cur[[1]]

# select HR and group into 2 subgroups seen
# cur %<>% dplyr::filter(comp == 'HR') %>%
#   mutate(subset = ifelse(ss.TotalScore > quantile(ss.TotalScore)[4], 'sub_hi', 'sub_lo'))

# table vars of interest - proportions
# prop.table(table(cur$subset, cur$timepoint), margin = 1) %>% `*`(100) %>% round(2) %>% barplot(beside=T, legend=rownames(.))
# prop.table(table(cur$subset, cur$dinuresp_old), margin = 1) %>% `*`(100) %>% round(2) %>% barplot(beside=T, legend=rownames(.))
# prop.table(table(cur$subset, cur$dinutx_ever), margin = 1) %>% `*`(100) %>% round(2) %>% barplot(beside=T, legend=rownames(.))
# prop.table(table(cur$subset, cur$site), margin = 1) %>% `*`(100) %>% round(2) %>% barplot(beside=T, legend=rownames(.))

# Raw values
# table(cur$dinutx_ever, cur$subset) %>% barplot(beside=T, legend=rownames(.))


```

### 306 gene signature for LR vs HR and LR/IR vs HR
```{r}

# Define comparison
dat <- nbl %>% dplyr::filter(group %in% c('HR', 'LR'))

# get PCA and boxplots
cur <- make_plots(
  DF = dat,
  UPGENES = sig306_up, 
  DOWNGENES = sig306_dn, 
  COMP = 'comp', 
  TIMEPOINT = 'dxprim', 
  STAT_TEST = 'wilcox.test',
  TITLE = '306gs: LR vs HR')

# plot 
cur[3]


# Define comparison
nbl$comp <- ifelse(nbl$group %in% c('LR', 'IR'), 'LR/IR', nbl$group)

# get PCA and boxplots
cur <- make_plots(
  DF = nbl,
  UPGENES = sig306_up, 
  DOWNGENES = sig306_dn, 
  COMP = 'comp', 
  TIMEPOINT = 'dxprim', 
  STAT_TEST = 'wilcox.test',
  TITLE = '306gs: LR/IR vs HR')

# plot 
cur[3]


```

### 306 gene signature for Dinutuxumab response among HR (1/2/3/4)
```{r}

# Define comparison
nbl$comp <- nbl$dinuresp_old
dat <- nbl %>% dplyr::filter(!comp %in% c('pending'))

# plot PCA and boxplots
cur <- make_plots(
  DF = dat,
  UPGENES = sig306_up, 
  DOWNGENES = sig306_dn, 
  COMP = 'comp', 
  TIMEPOINT = 'dxprim', 
  STAT_TEST = 'anova',
  TITLE = '306gs: Dinutux response among HR')

# plot and get df
cur[2]
cur[3]
#cur[4]

```

### 306 gene signature for Dinutuxumab response among HR (1/234)
```{r}

# Define comparison
nbl$comp <- nbl$dinuresp_old
nbl %<>% mutate(comp = ifelse(comp %in% c('group2', 'group3', 'group4'), 'group234', comp))
dat <- nbl %>% dplyr::filter(!comp %in% c('nothr','pending'))

# plot PCA and boxplots
cur <- make_plots(
  DF = dat,
  UPGENES = sig306_up, 
  DOWNGENES = sig306_dn, 
  COMP = 'comp', 
  TIMEPOINT = 'dxprim', 
  STAT_TEST = 'wilcox.test',
  TITLE = '306gs: Dinutux response 1v2/3/4')

# plot and get df
cur[2]
cur[3]
#cur[4]

```

### 306 gene signature for ever received Dinutuximab
```{r}

# Define comparison
nbl$comp <- nbl$dinutx_ever
dat <- nbl %>% mutate(comp = ifelse(comp == 1, '+Dinutux', '-Dinutux'))

# plot PCA and boxplots
cur <- make_plots(
  DF = dat,
  UPGENES = sig306_up, 
  DOWNGENES = sig306_dn, 
  COMP = 'comp', 
  TIMEPOINT = 'all', 
  STAT_TEST = 'wilcox.test',
  TITLE = '306gs: Ever recieved dinutux')

# plot and get df

cur[3]
#cur[4]

```

### 306 gene signature for dinutuximab during upfront maitenance therapy
```{r}

# Define comparison
nbl$comp <- nbl$ufmdinu
dat <- nbl 

# plot PCA and boxplots
cur <- make_plots(
  DF = dat,
  UPGENES = sig306_up, 
  DOWNGENES = sig306_dn, 
  COMP = 'comp', 
  TIMEPOINT = 'all', 
  STAT_TEST = 'anova',
  TITLE = '306gs: Dinutux during upfront mait.')

# plot and get df

cur[3]
#cur[4]

```

### 306 gene signature for tissue of origin
```{r}

# Define comparison
dat <- nbl %>%
  mutate(comp = ifelse(tissue_combined %in% c('bone', 'bone.marrow'), 'bone(marrow)', tissue_combined)) %>%
  mutate(comp = ifelse(tissue_combined %in% c('brain', 'cervical mass', 'cranial mass'), 'cranial/cervical', comp)) %>%
  mutate(comp = ifelse(tissue_combined %in% c('peritoneal mass', 'retroperitoneal mass'), '(retro)peritoneal', comp))
  
  

# plot PCA and boxplots
cur <- make_plots(
  DF = dat,
  UPGENES = sig306_up, 
  DOWNGENES = sig306_dn, 
  COMP = 'comp', 
  TIMEPOINT = 'all', 
  STAT_TEST = 'anova',
  TITLE = '306gs: By tissue of origin')

# plot and get df

cur[3]
#cur[4]

```


### 306 gene signature for rl.rf.pd (relapse/refractory dissease after upfront therapy)
```{r}

# Define comparison
nbl$comp <- nbl$rl.rf.pd
dat <- nbl%>% dplyr::filter(comp %in% c('Y', 'N'))

# plot PCA and boxplots
cur <- make_plots(
  DF = dat,
  UPGENES = sig306_up, 
  DOWNGENES = sig306_dn, 
  COMP = 'comp', 
  TIMEPOINT = 'all', 
  STAT_TEST = 'wilcox.test',
  TITLE = '306 gene sig')

# plot and get df

cur[3]
#cur[4]

```

### 306 gene signature for rl.rf.pd (relapse/refractory dissease after upfront therapy) in context of ufm dinutux
```{r}

#Of the patients who did not receive UFM dinu, how many had refractory, relapse or progressive disease?
#Of the patients who received UFM dinu, how many had refractory, relapse or progressive disease?


# Define comparison
nbl$comp <- paste0(nbl$ufmdinu, '_', nbl$rl.rf.pd)
dat <- nbl 

# plot PCA and boxplots
cur <- make_plots(
  DF = dat,
  UPGENES = sig306_up, 
  DOWNGENES = sig306_dn, 
  COMP = 'comp', 
  TIMEPOINT = 'all', 
  STAT_TEST = 'anova',
  TITLE = '306 gene sig')

# plot and get df

cur[3]
#cur[4]

```

### 306 gene signature for RLRF Dinutux treatment (how many developed refractory dx)
```{r}

# Within the group that was treated with RLRF Dinutuximab how many developed relapsed or refractory disease? 

# generate comparison
dat <- nbl[nbl$rl.rf.pd=="Y" & nbl$ufmdinu=="no",]
table(dat$rl.rf.pd, dat$rlrfdinu)
dat$comp <- dat$rlrfdinu

# plot PCA and boxplots
cur <- make_plots(
  DF = dat,
  UPGENES = sig306_up, 
  DOWNGENES = sig306_dn, 
  COMP = 'comp', 
  TIMEPOINT = 'all', 
  STAT_TEST = 'anova',
  TITLE = '306 gene sig')

# plot and get df

cur[3]
#cur[4]

```

### 306 gene signature Kaplan meijer for OS - 1st vs 4th quartile
```{r}

# Filter dat for those with OS scores, keeping non-duplicate patients
dat <- nbl %>% 
  # remove those w/o times
  dplyr::filter(!is.na(dinu_stime)) %>%
  # for duplicate preferentially keep the dxprims (doesn't really matter since all have same survival time vals)
  dplyr::mutate(timepoint = factor(timepoint, levels = c('dxprim', 'dx2prim', 'dxmet', 'posttx', 'rel_ref', 'pending'))) %>%
  arrange(timepoint) %>%
  dplyr::filter(!duplicated(patient))

# get sigscores for stratifying in kaplan meijer curve
cur <- make_plots(
  DF = dat,
  UPGENES = sig306_up, 
  DOWNGENES = sig306_dn, 
  COMP = 'comp', 
  TIMEPOINT = 'dxprim', 
  STAT_TEST = 'anova',
  TITLE = '306 gene sig')
cur <- cur[[1]]

# stratify by quartile 
cur %<>%
  mutate(comp = ifelse(ss.TotalScore >= quantile(ss.TotalScore)[3], 3, 2)) %>%
  mutate(comp = ifelse(ss.TotalScore >= quantile(ss.TotalScore)[4], 4, comp)) %>%
  mutate(comp = ifelse(ss.TotalScore < quantile(ss.TotalScore)[2], 1, comp)) 
cur %<>% dplyr::filter(comp %in% c(1,4))
#cur %<>% mutate(comp = ifelse(comp %in% c(1,2,3), 123, 4))

# stratify by visual cutoff
#cur %<>% mutate(comp = ifelse(ss.TotalScore > -0.05, 'hi', 'lo'))

# plot Kaplan meijer
km_fit <- survfit(Surv(dinu_stime, rep(1, length(cur$rlib))) ~ comp, data=cur)
summary(km_fit, times = c(1,30,60,90*(1:10)))
autoplot(km_fit)

```

### 306 gene signature Kaplan meijer for EFS
```{r}

# Filter dat for those with OS scores, keeping non-duplicate patients
dat <- nbl %>% 
  # remove those w/o times
  dplyr::filter(!is.na(dinu_efstime)) %>%
  # for duplicate preferentially keep the dxprims (doesn't really matter since all have same survival time vals)
  dplyr::mutate(timepoint = factor(timepoint, levels = c('dxprim', 'dx2prim', 'dxmet', 'posttx', 'rel_ref', 'pending'))) %>%
  arrange(timepoint) %>%
  dplyr::filter(!duplicated(patient))

# get sigscores for stratifying in kaplan meijer curve
cur <- make_plots(
  DF = dat,
  UPGENES = sig306_up, 
  DOWNGENES = sig306_dn, 
  COMP = 'comp', 
  TIMEPOINT = 'dxprim', 
  STAT_TEST = 'anova',
  TITLE = '306 gene sig')
cur <- cur[[1]]

# stratify by quartile 
cur %<>%
  mutate(comp = ifelse(ss.TotalScore >= quantile(ss.TotalScore)[3], 3, 2)) %>%
  mutate(comp = ifelse(ss.TotalScore >= quantile(ss.TotalScore)[4], 4, comp)) %>%
  mutate(comp = ifelse(ss.TotalScore < quantile(ss.TotalScore)[2], 1, comp)) 
cur %<>% dplyr::filter(comp %in% c(1,4))
#cur %<>% mutate(comp = ifelse(comp %in% c(1,2,3), 123, 4))

# stratify by visual cutoff
#cur %<>% mutate(comp = ifelse(ss.TotalScore > -0.05, 'hi', 'lo'))

# plot Kaplan meijer
km_fit <- survfit(Surv(dinu_efstime, rep(1, length(cur$rlib))) ~ comp, data=cur)
summary(km_fit, times = c(1,30,60,90*(1:10)))
autoplot(km_fit)

```


## 63 gene signature

### Load gene signature
```{r}
sig63 <- fread('/mctp/share/users/rebernrj/projects/ferroptosisGeneSig/out/gene_signatures/erastin_clue24h_gse104462_63genes.txt',
                data.table = F)

# Upregulated
sig63_up <- sig63 %>% 
  dplyr::filter(upregulated == T) %>% 
  dplyr::select(V1) %>% unlist()

# Downregulated
sig63_dn <- sig63 %>% 
  dplyr::filter(upregulated == F) %>% 
  dplyr::select(V1) %>% unlist()


```


## 46 gene signature

### Load gene signature
```{r}
sig46 <- fread('/mctp/share/users/rebernrj/projects/ferroptosisGeneSig/out/gene_signatures/erastin_clue24h_gse104462_46genes.txt',
                data.table = F)

# Upregulated
sig46_up <- sig46 %>% 
  dplyr::filter(upregulated == T) %>% 
  dplyr::select(V1) %>% unlist()

# Downregulated
sig46_dn <- sig46 %>% 
  dplyr::filter(upregulated == F) %>% 
  dplyr::select(V1) %>% unlist()


```


## 96 gene signature

### Load gene signature
```{r}
sig96 <- fread('/mctp/share/users/rebernrj/projects/ferroptosisGeneSig/out/gene_signatures/erastin_elife_correlation_gse104462_96genes.txt',
                data.table = F)

# Upregulated
sig96_up <- sig96 %>% 
  dplyr::filter(GSE104462_LFC > 0) %>% 
  dplyr::select(V1) %>% unlist()

# Downregulated
sig96_dn <- sig96 %>% 
  dplyr::filter(GSE104462_LFC < 0) %>% 
  dplyr::select(V1) %>% unlist()


```


## Elife gene signature

## Load gene signature
```{r}

elife_up <- c('CHAC1', 'DDIT4', 'LOC284561', 'ASNS', 'TSC22D3', 'DDIT3', 'JDP2', 'SESN2',
              'SLC1A4', 'PCK2', 'SLC7A11', 'TXNIP', 'VLDLR', 'GPT2', 'PSAT1', 'C9ORF150',
              'SLC7A5', 'HERPUD1', 'XBP1', 'ATF3', 'SLC3A2', 'CBS', 'ATF4', 'ZNF419', 
              'KLHL24', 'TRIB3', 'ZNF643', 'ATP6V1G2', 'VEGFA', 'GDF15', 'TUBE1', 
              'ARRDC3', 'CEBPG')

elife_down <- c('SNORA16A', 'RGS4', 'MUTED-TXNDC5', 'LOC390705')
```

### Correlation of elife score with 306 gene score
Correlation b/t scores is poor (<.1)
Different samples are in the high scoring groups of both scores. Only 11/25 samples are in the high scoring groups of both
```{r}

# Define comparison
nbl$comp <- 'all'

# get elife values
cur <- make_plots(
  DF = nbl,
  UPGENES = elife_up, 
  DOWNGENES = elife_down, 
  COMP = 'comp', 
  TIMEPOINT = 'all', 
  STAT_TEST = 'anova',
  TITLE = 'elife gene sig')
cur <- cur[[1]]
cur %<>% dplyr::select(rlib, ss.TotalScore, directional_Zsum) %>% set_colnames(c('rlib', 'elife.ss', 'elife.zsum'))

# get sig306 values
temp <- make_plots(
  DF = nbl,
  UPGENES = sig306_up, 
  DOWNGENES = sig306_dn, 
  COMP = 'comp', 
  TIMEPOINT = 'all', 
  STAT_TEST = 'anova',
  TITLE = 'elife gene sig')
temp <- temp[[1]]
temp %<>% dplyr::select(rlib, ss.TotalScore, directional_Zsum) %>% set_colnames(c('rlib', 'sig306.ss', 'sig306.zsum'))

# merge
cur %<>% merge(., temp, by = 'rlib')

# Check correlation
## singscore elife vs singscore 306
ggscatter(cur, x='elife.ss', y='sig306.ss',
          add='reg.line',
          add.params = list(color='blue', fill='lightgray'),
          conf.int = T) +
  stat_cor(method = "pearson")
## directional zsum elife vs directional zsum sig306
ggscatter(cur, x='elife.zsum', y='sig306.zsum',
          add='reg.line',
          add.params = list(color='blue', fill='lightgray'),
          conf.int = T) +
  stat_cor(method = "pearson")

# See if they identify the same samples as high scoring
cur %<>% 
  mutate(elife_sub = ifelse(elife.ss >= quantile(elife.ss)[4], 'elife_hi', 'elife_lo')) %>%
  mutate(sig306_sub = ifelse(sig306.ss >= quantile(sig306.ss)[4], 'sig306_hi', 'sig306_lo'))
table(cur$elife_sub, cur$sig306_sub)

```

### Correlation of elife score with T cell proportion
```{r}

# Load T cell data
epic <- fread('/mctp/share/users/rebernrj/projects/ferroptosisGeneSig/out/EPIC/NBL/cell_proportions/nbl_epic_proportions_all.txt', data.table = F) %>%
  dplyr::select(rlib, Bcells:otherCells)
tcr <- fread('/mctp/share/users/rebernrj/projects/ferroptosisGeneSig/data/nbl_cohort/other/nbl-crisp-canio_tcr-summary.csv', data.table = F)


# Load gene sig scores
cur <- make_plots(
  DF = nbl,
  UPGENES = elife_up, 
  DOWNGENES = elife_down, 
  COMP = 'comp', 
  TIMEPOINT = 'all', 
  STAT_TEST = 'anova',
  TITLE = '306 gene sig')
cur <- cur[[1]]

# Combine T cell data with gene signature 
comb <- unique(merge(cur, epic, by = 'rlib', all.x = T))
comb <- merge(comb, tcr, by.x = 'rlib', by.y='sid', all.x=T)

# T cell clones
ggscatter(comb, x='ss.TotalScore', y='clones_T',
          add='reg.line',
          add.params = list(color='blue', fill='lightgray'),
          conf.int = T) +
  stat_cor(method = "pearson")

# B cell clones
ggscatter(comb, x='ss.TotalScore', y='clones_B',
          add='reg.line',
          add.params = list(color='blue', fill='lightgray'),
          conf.int = T) +
  stat_cor(method = "pearson")

# T cell Gini
ggscatter(comb, x='ss.TotalScore', y='gini_T',
          add='reg.line',
          add.params = list(color='blue', fill='lightgray'),
          conf.int = T) +
  stat_cor(method = "pearson")

# B cell Gini
ggscatter(comb, x='ss.TotalScore', y='gini_B',
          add='reg.line',
          add.params = list(color='blue', fill='lightgray'),
          conf.int = T) +
  stat_cor(method = "pearson")

# CD4
ggscatter(comb, x='ss.TotalScore', y='CD4_Tcells',
          add='reg.line',
          add.params = list(color='blue', fill='lightgray'),
          conf.int = T) +
  stat_cor(method = "pearson")

# CD8
ggscatter(comb, x='ss.TotalScore', y='CD8_Tcells',
          add='reg.line',
          add.params = list(color='blue', fill='lightgray'),
          conf.int = T) +
  stat_cor(method = "pearson")


```


# Look at expression of cell markers

### NK cell marker expression
```{r}

# get NK cell genes from literature
nk <- fread('/mctp/share/users/rebernrj/projects/ferroptosisGeneSig/data/doi_10.1158_2326-6966.CIR-18-0500_NKcellGeneSig/s1.txt', data.table = F, skip = 2)
nk %<>% dplyr::select(`HGNC Symbol`, `Cursons\nGuimaraes\nsigGene`, logFC) %>%
  set_colnames(c('gene', 'sig', 'logfc'))
## upregulated
nk_up <- nk %>% dplyr::filter(logfc > 0) %>%
  dplyr::filter(sig == T) %>%
  dplyr::select(gene) %>%
  unlist() %>%
  set_names(NULL)


# select variables needed for plot and format properly
toPlot <- nbl %>% 
  mutate(comp = ifelse(dinutx_ever == 1, 'Yes', 'No')) %>%
  dplyr::select(comp, all_of(nk_up)) %>%
  gather(., gene, expression, CD160:XCL2) %>%
  mutate(log10exp = log10(expression+0.001))

# Plot 
toPlot %>% ggplot(aes(x=gene, y=log10exp)) + 
    geom_boxplot(outlier.shape = NA, color='black') +
    geom_jitter(alpha=0.6, width=0.15) +
    theme_bw(base_size=16) +
    labs(subtitle='NK cell gene expresion (TPM)') +
    xlab(NULL) +
    ylab('log10(exp + 0.001)') +
    theme(axis.text.x = element_text(angle = 90))


```










