---
title: "eda_geneSignature"
author: "Ryan Rebernick"
date: "3/29/2021"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

# This parameter is needed for installation of org.Hs.eg.db
Sys.setenv(R_INSTALL_STAGED = FALSE)

```


# Description:
This script creates gene signatures for ferroptosis/erastin exposure using several publically available datasets including: 
1) Pharmacological inhibition of cystine-glutamate exchange induces endoplasmic reticulum stress and ferroptosis. doi: 10.7554/eLife.02523.
2) Ferroptotic cell death triggered by conjugated linolenic acids is mediated by ACSL1. (GSE162069)
3) Ferroptosis is governed by differential regulation of transcription in liver cancer. (GSE104462) PMID: 31108460
4) Clue.io data for erastin exposure (6h/24h post exposure). Output is Z scores from L1000 assay. https://clue.io/ 

To accomplish this, the 24h Clue erastin data is used as the basis for the signatures which are then narrowed down by signficance/presence in RNA-seq datasets as well as consistent directional changes throughout all datasets. These gene signatures are then used to plot PCA of the samples and GO analysis is also performed on the signatures to rule out confounding processes like apoptosis. 

# Load packages
```{r, warning=F, message=FALSE}

library('plyr')
library('dplyr')
library('stringr')
library('data.table')
library('tidyverse')
library('magrittr')

library('ComplexHeatmap')
library(circlize)
library('cowplot')
library('ggfortify')
#devtools::install_github("yanlinlin82/ggvenn")
library(ggvenn)

```

# Color scheme
```{r}

# color scale for RNA-seq heatmaps
col_fun = colorRamp2(c(0, 4), c("white", "red"))

```


# Find DEGs in common between 3 RNA-seq datasets
1) Pharmacological inhibition of cystine-glutamate exchange induces endoplasmic reticulum stress and ferroptosis. doi: 10.7554/eLife.02523.
2) Ferroptotic cell death triggered by conjugated linolenic acids is mediated by ACSL1. (GSE162069)
3) Ferroptosis is governed by differential regulation of transcription in liver cancer. (GSE104462) PMID: 31108460


## Load DEGs
GSE152069 and GSE104462 were re-analyzed using ensembl v97 assembly. The elife paper only provided fpkm and this was converted to counts. To convert to counts, gene length values were compressed from ensembl 67 - one of the likely assemblies used with the paper based on date. The authors from the elife paper do not deem it pertinent to include which assembly was used. Additionally, for the elife paper DEGs to broaden the number of genes, a fold change > 2 was considered significant regardless of p-value. This was done because a) they did it in the paper, and b) I was having trouble recapitulating their findings using only signfiicant genes.
```{r}

# elife 
elife <- fread('/mctp/share/users/rebernrj/projects/ferroptosisGeneSig/int/elife2014_erastin/rna_seq/elife_degs_erastin_control.txt', data.table = F)

# GSE104462 (erastin)
gse104462 <- fread('/mctp/share/users/rebernrj/projects/ferroptosisGeneSig/int/GSE104462/deseq2/GSE104462_degs_erastin_control.txt', data.table = F)

# GSE162069 (ML162)
gse162069 <- fread('/mctp/share/users/rebernrj/projects/ferroptosisGeneSig/int/GSE162069/deseq2/GSE162069_degs_erastin_control.txt', data.table = F)


```

# Lets prep the clue data so we can see how our genes of interest will look in this dataset.
Hint: It won't be clean....

## Function to clean clue data
```{r}

clean_clue <- function(clue, dose = '10 µM', time = '6 h', see_avail_opts = T){
  
  # If fxn allows for viewing avail options for timepoints/concentrations
  if (see_avail_opts == T) {
    # print available cell lines and doses
    temp <- as.data.frame(t(unname(clue[c(1,5),10:ncol(clue)])))
    colnames(temp) <- c('line', 'dose')
    print(table(temp$line, temp$dose))
    
        # print available time points and doses
    temp <- as.data.frame(t(unname(clue[5:6,10:ncol(clue)])))
    colnames(temp) <- c('dose', 'time')
    print(table(temp$dose, temp$time))
  }
  
  else {
    # keep time point of interest
    if(!is.null(time)){
      clue =  clue[, c(rep(T, 9) ,clue[6, 10:ncol(clue)] == time)]
    }
    
    # keep dose of interest
    if(!is.null(dose)){
    clue =  clue[, c(rep(T, 9) ,clue[5, 10:ncol(clue)] == dose)]
    }
    
    # separate metadata from z scores for duplicate removal
    temp <- clue[, 10:ncol(clue)] # z scores
    clue <- clue[,1:9] # meta
    
    # For duplicates, keep the replicate with a higher distil_cc_q75
    # This is the replicate consistency score (spearman correlation between replicates of lvl 4 profile)
    # https://clue.io/connectopedia/category/Analytical%20Methods
    ## order z scores by the distil_cc_q75 to facilitate removal of lower distil value
    temp <- temp[,rev(order(temp[7,]))]
    
    # recombine z scores with metadata
    clue[,10:(10+length(temp))] <- temp
    
    # Rename columns with cell line
    colnames(clue)[10:ncol(clue)] <- clue[1,10:ncol(clue)]
    
    # remove duplicate columns, keeping the one with the higher distil_cc_q75 score (higher corr between reps)
    # works because ordered by distil score above
    clue <- clue[,!duplicated(colnames(clue))]
    
    # elimiante unecessary meta data
    clue <- clue[9:length(clue$id),]
    
    # convert to numeric
    clue[10:ncol(clue)] <- apply(clue[10:ncol(clue)], 2, as.numeric)
    
    # select cols to keep for correlation
    clue %<>% dplyr::select(-id, -pr_gene_title, -pr_is_lm, -pr_is_bing, -self_correlation, -frac_self_rank, -p_value, -gene_space)
    
    return(clue)
  }
}

```

## Clean clue data from cells exposed to erastin
stratify by timepoint and dosage
```{r}

# clue erastin - loaded as 2 seperate files because of export limits of clue.io
clue.erastin <- fread('/mctp/share/users/rebernrj/projects/ferroptosisGeneSig/data/clue/erastin/clue_v1.3_erastin_1.gct', skip = 2, data.table = F)
temp <- fread('/mctp/share/users/rebernrj/projects/ferroptosisGeneSig/data/clue/erastin/clue_v1.3_erastin_2.gct', skip = 2, data.table = F)

# merge clue erastin files
clue.erastin[,90:97] <- temp[,10:17]

# see options
clean_clue(clue.erastin, see_avail_opts = T)

# 5um, 6h
clue.erastin_6h_5um <- clean_clue(clue.erastin, dose = '5 µM', time = '6 h', see_avail_opts = F)

# 5um, 24h
clue.erastin_24h_5um <- clean_clue(clue.erastin, dose = '5 µM', time = '24 h', see_avail_opts = F)

# 10um, 24h
clue.erastin_24h_10um <- clean_clue(clue.erastin, dose = '10 µM', time = '24 h', see_avail_opts = F)

```

# Focus on 24 hour time point to build signature working backwards (all Clue genes)
## Find the genes significant in the 24 hour data set (clue)
Will start with all of the clue genes (including imputed).
To do this:
1) merge both 24h clue datasets together (5um/10um)
2) Select genes which are changing the same direction in at least 10/13 samples
3) Filter the remaining genes to have an abs(sum of the z scores for the gene) > mean(abs(sum of the z scores for the gene))
```{r}

# merge 24h clue data frames adding dosage info to colnames
## 10um
clue.erastin_24h_10um %<>% set_rownames(clue.erastin_24h_10um$pr_gene_symbol) %>%
  dplyr::select(-pr_gene_symbol)
colnames(clue.erastin_24h_10um) <- paste0(colnames(clue.erastin_24h_10um), '_10um')
## 5um
clue.erastin_24h_5um %<>% set_rownames(clue.erastin_24h_5um$pr_gene_symbol) %>%
  dplyr::select(-pr_gene_symbol)
colnames(clue.erastin_24h_5um) <- paste0(colnames(clue.erastin_24h_5um), '_5um')
## merge
clue.erastin_24h <- merge(clue.erastin_24h_5um, clue.erastin_24h_10um, by = 0)
clue.erastin_24h %<>% set_rownames(clue.erastin_24h$Row.names) %>%
  dplyr::select(-Row.names)

# Filter for those changing the same way in at least 10/13
clue.erastin_24h %<>% 
  dplyr::mutate(sums = (HA1E_5um > 0) + 
                  (A375_5um > 0) +
                  (HT29_5um > 0) +
                  (MCF7_5um > 0) +
                  (PC3_5um > 0) +
                  (A549_5um > 0) +
                  (HCC515_5um > 0) +
                  (VCAP_5um > 0) +
                  (HEPG2_10um > 0) +
                  (A375_10um > 0) +
                  (HCC515_10um > 0) +
                  (A549_10um > 0) +
                  (HA1E_10um > 0)) %>%
  dplyr::mutate(upregulated = ifelse(sums >= 8, T, F))
clue.erastin_24h %<>% dplyr::filter(sums >= 10 | sums <= 3) %>%
  dplyr::select(-sums)

# Find most significantly changing genes (summation of z scores)
clue.erastin_24h %<>%
  mutate(sums = rowSums(abs(clue.erastin_24h[,-14]))) 
clue.erastin_24h %<>% dplyr::filter(sums >= mean(clue.erastin_24h$sums)) 
clue.erastin_24h %<>% dplyr::select(-sums)


```

## see how many of the genes significant from the 24h clue dataset recapitulate in the good RNA-seq data set
1) Merge GSE104462 with the signficant 24h Clue RNA-seq genes
2) Filter for genes changing the same direction in both datasets
3) Filter for genes with a padj < 0.1 in the RNA-seq dataset
```{r}
# GSE104462 (erastin)
gse104462 <- fread('/mctp/share/users/rebernrj/projects/ferroptosisGeneSig/int/GSE104462/deseq2/GSE104462_degs_erastin_control.txt', data.table = F)

# get desired values for gse104462
gse104462 %<>%
  set_rownames(gse104462$Row.names) %>%
  dplyr::select(log2FoldChange, padj)

# Find genes present in both data sets
sig_24h <- merge(clue.erastin_24h, gse104462, by = 0)

# Filter for those changing the same way in both
sig_24h %<>% dplyr::filter((upregulated == T & log2FoldChange > 0) | (upregulated == F & log2FoldChange < 0))

# filter for those significant in the RNA-seq data set
sig_24h %<>% dplyr::filter(padj < 0.1)

# clean for processing
sig_24h %<>% set_rownames(sig_24h$Row.names) %>%
  dplyr::select(upregulated:padj, HA1E_5um:HA1E_10um)
colnames(sig_24h)[2:3] <- c('GSE104462_LFC', 'GSE104462_padj')
sig_24h %<>% dplyr::arrange(GSE104462_LFC)


```

## Clean all datasets for plot of 24h significant genes in all data points

#### Load and merge RNA-seq data sets and add rownames to all datasets 
```{r}

# elife 
elife <- fread('/mctp/share/users/rebernrj/projects/ferroptosisGeneSig/int/elife2014_erastin/rna_seq/elife_degs_erastin_control.txt', data.table = F)

# GSE104462 (erastin)
gse104462 <- fread('/mctp/share/users/rebernrj/projects/ferroptosisGeneSig/int/GSE104462/deseq2/GSE104462_degs_erastin_control.txt', data.table = F)

# GSE162069 (ML162)
gse162069 <- fread('/mctp/share/users/rebernrj/projects/ferroptosisGeneSig/int/GSE162069/deseq2/GSE162069_degs_erastin_control.txt', data.table = F)


# merge elife and gse104462
rna <- merge(gse104462, elife, by = 'Row.names', all.x = T)
rna %<>% dplyr::select(Row.names, log2FoldChange.x, log2FoldChange.y, padj.x, padj.y)
colnames(rna) <- c('Row.names', 'GSE104462_LFC', 'elife_LFC' , 'GSE104462_padj', 'padj.elife')

# merge GSE162069
rna <- merge(rna, gse162069, all.x = T)
rna %<>% dplyr::select(Row.names: elife_LFC, log2FoldChange, GSE104462_padj:padj.elife, padj)
colnames(rna) <- c('Row.names', 'GSE104462_LFC', 'elife_LFC', 'GSE162069_LFC', 'GSE104462_padj', 'elife_padj', 'GSE162069_padj')

# add rownames to RNA-seq
## RNA-seq
rna %<>% set_rownames(rna$Row.names) %<>% dplyr::select(-Row.names)

# Add rownames to Clue
## 6h 5um
clue.erastin_6h_5um %<>% set_rownames(clue.erastin_6h_5um$pr_gene_symbol) %>% dplyr::select(-pr_gene_symbol)
## already set for the 24h time points

```

## Select sig_24h genes and order for heatmap
Get the genes from the 24h signature in all of the available datasets and order the corresponding datasets for plotting. ComplexHeatmap should automatically order by Row.names but it's being dumb and throwing an error if they're not ordered. 
```{r}

# select genes
rna.sig_24h <- rna %>% dplyr::filter(rownames(.) %in% rownames(sig_24h))
clue.erastin_6h_5um.sig_24h <- clue.erastin_6h_5um %>% dplyr::filter(rownames(.) %in% rownames(sig_24h))
clue.erastin_24h_10um.sig_24h <- clue.erastin_24h_10um %>% dplyr::filter(rownames(.) %in% rownames(sig_24h)) 
clue.erastin_24h_5um.sig_24h <- clue.erastin_24h_5um %>% dplyr::filter(rownames(.) %in% rownames(sig_24h))

# order
rna.sig_24h <- rna.sig_24h[order(match(rownames(rna.sig_24h),rownames(sig_24h))),]
clue.erastin_6h_5um.sig_24h <- clue.erastin_6h_5um.sig_24h[order(match(rownames(clue.erastin_6h_5um.sig_24h),rownames(sig_24h))),]
clue.erastin_24h_10um.sig_24h <- clue.erastin_24h_10um.sig_24h[order(match(rownames(clue.erastin_24h_10um.sig_24h),rownames(sig_24h))),]
clue.erastin_24h_5um.sig_24h <- clue.erastin_24h_5um.sig_24h[order(match(rownames(clue.erastin_24h_5um.sig_24h),rownames(sig_24h))),]

```


## Plot gene signature in all data sets
Row ordering is based on clustering algorithm
```{r}

# row split
up <- table(rna.sig_24h$GSE104462_LFC > 0)[2]
down <- table(rna.sig_24h$GSE104462_LFC > 0)[1]

set.seed(11)
Heatmap(rna.sig_24h[,1:3],
        show_column_dend = F,
        show_row_dend = F,
        show_row_names = F,
        column_names_gp = gpar(fontsize = 7),
        column_title = 'RNA-seq',
        column_title_gp = gpar(fontsize = 8),
        show_heatmap_legend = F,
        row_split = c(rep('down', down), rep('up', up)),
        na_col = 'black',
        cluster_rows = T) +
  
Heatmap(clue.erastin_6h_5um.sig_24h,
        show_column_dend = F,
        show_row_dend = F,
        show_row_names = F,
        row_names_gp = gpar(fontsize = 5),
        column_names_gp = gpar(fontsize = 7),
        column_title = 'Clue-Erastin 5um-6h',
        column_title_gp = gpar(fontsize = 8),
        show_heatmap_legend = F,
        clustering_method_columns = 'complete') +
  
Heatmap(clue.erastin_24h_5um.sig_24h,
        show_column_dend = F,
        show_row_dend = F,
        show_row_names = F,
        column_names_gp = gpar(fontsize = 7),
        column_title = '5um-24h',
        column_title_gp = gpar(fontsize = 8),
        show_heatmap_legend = F) +
  
  Heatmap(clue.erastin_24h_10um.sig_24h,
        show_column_dend = F,
        show_row_dend = F,
        show_row_names = F,
        column_names_gp = gpar(fontsize = 7),
        column_title = '10um-24h',
        column_title_gp = gpar(fontsize = 8),
        show_heatmap_legend = F)

```

## order the genes significant from the 24h time point by the 6h samples (significant in the most genes)
Sort by the number of genes changing the same way w/in the upregulated genes for the 6h dataset.
```{r}

# Order clue.erastin_6h_5um.sig_24h based on erastin 6h and upreg/downreg status
clue.erastin_6h_5um.sig_24h$upregulated <- sig_24h$upregulated
clue.erastin_6h_5um.sig_24h$sums <- 0
for(x in 1:length(clue.erastin_6h_5um.sig_24h$NCIH508)){
  clue.erastin_6h_5um.sig_24h$sums[x] <-table(clue.erastin_6h_5um.sig_24h[x,] > 0.5)[2]
}
clue.erastin_6h_5um.sig_24h %<>% arrange(upregulated, -sums)
clue.erastin_6h_5um.sig_24h %<>% dplyr::select(-sums, -upregulated)

# order other datasets based on clue.erastin_6h_5um.sig_24h
rna.sig_24h <- rna.sig_24h[order(match(rownames(rna.sig_24h),rownames(clue.erastin_6h_5um.sig_24h))),]
clue.erastin_24h_10um.sig_24h <- clue.erastin_24h_10um.sig_24h[order(match(rownames(clue.erastin_24h_10um.sig_24h),rownames(clue.erastin_6h_5um.sig_24h))),]
clue.erastin_24h_5um.sig_24h <- clue.erastin_24h_5um.sig_24h[order(match(rownames(clue.erastin_24h_5um.sig_24h),rownames(clue.erastin_6h_5um.sig_24h))),]


```

## Plot gene signature in all data sets using ordering based on the 6h samples 
Clustering of the 6h samples looks slightly better.
```{r}

# row split
up <- table(rna.sig_24h$GSE104462_LFC > 0)[2]
down <- table(rna.sig_24h$GSE104462_LFC > 0)[1]

set.seed(11)
Heatmap(rna.sig_24h[,1:3],
        show_column_dend = F,
        show_row_dend = F,
        show_row_names = F,
        column_names_gp = gpar(fontsize = 7),
        column_title = 'RNA-seq',
        column_title_gp = gpar(fontsize = 8),
        show_heatmap_legend = F,
        row_split = c(rep('down', down), rep('up', up)),
        na_col = 'black',
        cluster_rows = F,
        cluster_columns = T) +
  
Heatmap(clue.erastin_6h_5um.sig_24h,
        show_column_dend = F,
        show_row_dend = F,
        show_row_names = F,
        row_names_gp = gpar(fontsize = 5),
        column_names_gp = gpar(fontsize = 7),
        column_title = 'Clue-Erastin 5um-6h',
        column_title_gp = gpar(fontsize = 8),
        show_heatmap_legend = F,
        clustering_method_columns = 'complete') +
  
Heatmap(clue.erastin_24h_5um.sig_24h,
        show_column_dend = F,
        show_row_dend = F,
        show_row_names = F,
        column_names_gp = gpar(fontsize = 7),
        column_title = '5um-24h',
        column_title_gp = gpar(fontsize = 8),
        show_heatmap_legend = F) +
  
  Heatmap(clue.erastin_24h_10um.sig_24h,
        show_column_dend = F,
        show_row_dend = F,
        show_row_names = F,
        column_names_gp = gpar(fontsize = 7),
        column_title = '10um-24h',
        column_title_gp = gpar(fontsize = 8),
        show_heatmap_legend = F)

```

## Order based on 24h samples and plot
```{r}

## order by 24h row sums
clue.erastin_24h_5um.sig_24h %<>% dplyr::mutate(sums = rowSums(abs(clue.erastin_24h_5um.sig_24h))) 
## label as upreg/downreg
clue.erastin_24h_5um.sig_24h$upregulated <- sig_24h$upregulated
clue.erastin_24h_5um.sig_24h %<>%
  arrange(upregulated, -sums) %>%
  dplyr::select(-sums, -upregulated)

# order other datasets based on clue.erastin_6h_5um.sig_24h
rna.sig_24h <- rna.sig_24h[order(match(rownames(rna.sig_24h),rownames(clue.erastin_24h_5um.sig_24h))),]
clue.erastin_6h_5um.sig_24h <- clue.erastin_6h_5um.sig_24h[order(match(rownames(clue.erastin_6h_5um.sig_24h),rownames(clue.erastin_24h_5um.sig_24h))),]
clue.erastin_24h_10um.sig_24h <- clue.erastin_24h_10um.sig_24h[order(match(rownames(clue.erastin_24h_10um.sig_24h),rownames(clue.erastin_24h_5um.sig_24h))),]

```

## Plot gene signature in all data sets using ordering based on the 6h samples 
Clustering appears the best in this one
```{r}

# row split
up <- table(rna.sig_24h$GSE104462_LFC > 0)[2]
down <- table(rna.sig_24h$GSE104462_LFC > 0)[1]

set.seed(11)
Heatmap(rna.sig_24h[,1:3],
        show_column_dend = F,
        show_row_dend = F,
        show_row_names = F,
        column_names_gp = gpar(fontsize = 7),
        column_title = 'RNA-seq',
        column_title_gp = gpar(fontsize = 8),
        show_heatmap_legend = F,
        row_split = c(rep('down', down), rep('up', up)),
        na_col = 'black',
        cluster_rows = F,
        cluster_columns = T) +
  
Heatmap(clue.erastin_6h_5um.sig_24h,
        show_column_dend = F,
        show_row_dend = F,
        show_row_names = F,
        row_names_gp = gpar(fontsize = 5),
        column_names_gp = gpar(fontsize = 7),
        column_title = 'Clue-Erastin 5um-6h',
        column_title_gp = gpar(fontsize = 8),
        show_heatmap_legend = F,
        clustering_method_columns = 'complete') +
  
Heatmap(clue.erastin_24h_5um.sig_24h,
        show_column_dend = F,
        show_row_dend = F,
        show_row_names = F,
        column_names_gp = gpar(fontsize = 7),
        column_title = '5um-24h',
        column_title_gp = gpar(fontsize = 8),
        show_heatmap_legend = F) +
  
  Heatmap(clue.erastin_24h_10um.sig_24h,
        show_column_dend = F,
        show_row_dend = F,
        show_row_names = F,
        column_names_gp = gpar(fontsize = 7),
        column_title = '10um-24h',
        column_title_gp = gpar(fontsize = 8),
        show_heatmap_legend = F)

```

## Subset sig_24h further based on subclustering
#### Combine all clue datasets into single df to improve clustering and subset of interesting genes
Separating the heatmaps by origin (RNA-seq vs Clue) makes the data easier to visualize but impairs the ability to subcluster on groups besides the leftmost heatmap plotted. By combining the clue dataframes (6h and 24h) it makes it easier to subset out interesting clusters.
```{r}

# change colnames to make it easier to distinguish cell lines that appear in all time points
colnames(clue.erastin_6h_5um.sig_24h) <- paste0(colnames(clue.erastin_6h_5um.sig_24h), '_5um_6h')
colnames(clue.erastin_24h_5um.sig_24h) <- paste0(colnames(clue.erastin_24h_5um.sig_24h), '_24h')
colnames(clue.erastin_24h_10um.sig_24h) <- paste0(colnames(clue.erastin_24h_10um.sig_24h), '_24h')

# merge 6h and 24h clue data frames
sig_24h_merge <- merge(clue.erastin_6h_5um.sig_24h, clue.erastin_24h_5um.sig_24h, by = 0)
sig_24h_merge %<>% set_rownames(sig_24h_merge$Row.names) %>% dplyr::select(-Row.names)
sig_24h_merge <- merge(sig_24h_merge, clue.erastin_24h_10um.sig_24h, by = 0)
sig_24h_merge %<>% set_rownames(sig_24h_merge$Row.names) %>% dplyr::select(-Row.names)

# Plot
set.seed(11)
Heatmap(sig_24h_merge,
        show_column_dend = F,
        show_row_dend = F,
        show_row_names = F,
        column_names_gp = gpar(fontsize = 7),
        column_title = 'Combined Clue data',
        column_title_gp = gpar(fontsize = 8),
        show_heatmap_legend = F,
        row_km = 4,
        column_km = 3,,
        row_km_repeats = 100,
        na_col = 'black',
        cluster_rows = T,
        cluster_columns = T)


# remove lines with strong signal as these impeded further clustering 
temp <- c('T3M10_5um_6h', 'CORL23_5um_6h', 'SNU1040_5um_6h', 'CL34_5um_6h', 'NCIH1694_5um_6h')
sig_24h_merge_sub <- sig_24h_merge %>% dplyr::select(-temp)

# Plot without strong cell lines
set.seed(11)
H <- Heatmap(sig_24h_merge_sub,
        show_column_dend = F,
        show_row_dend = F,
        show_row_names = F,
        column_names_gp = gpar(fontsize = 7),
        column_title = 'Combined Clue data',
        column_title_gp = gpar(fontsize = 8),
        show_heatmap_legend = F,
        row_km = 5,
        column_km = 3,
        row_km_repeats = 100,
        na_col = 'black',
        cluster_rows = T,
        cluster_columns = T)
H <- draw(H)

# Take cluster 1 and 5
ComplexHeatmap::row_order(H)
sig_24h_sub <- rownames(sig_24h_merge_sub[c(row_order(H)[['5']],row_order(H)[['1']]),])
sig_24h_sub <- sig_24h %>% dplyr::filter(rownames(.) %in% sig_24h_sub)

```

#### Re-plot subset of sig_24h
```{r}

# select genes
rna.sig_24h_sub <- rna %>% dplyr::filter(rownames(.) %in% rownames(sig_24h_sub))
clue.erastin_6h_5um.sig_24h_sub <- clue.erastin_6h_5um %>% dplyr::filter(rownames(.) %in% rownames(sig_24h_sub))
clue.erastin_24h_10um.sig_24h_sub <- clue.erastin_24h_10um %>% dplyr::filter(rownames(.) %in% rownames(sig_24h_sub)) 
clue.erastin_24h_5um.sig_24h_sub <- clue.erastin_24h_5um %>% dplyr::filter(rownames(.) %in% rownames(sig_24h_sub))

# order
rna.sig_24h_sub <- rna.sig_24h_sub[order(match(rownames(rna.sig_24h_sub),rownames(sig_24h_sub))),]
clue.erastin_6h_5um.sig_24h_sub <- clue.erastin_6h_5um.sig_24h_sub[order(match(rownames(clue.erastin_6h_5um.sig_24h_sub),rownames(sig_24h_sub))),]
clue.erastin_24h_10um.sig_24h_sub <- clue.erastin_24h_10um.sig_24h_sub[order(match(rownames(clue.erastin_24h_10um.sig_24h_sub),rownames(sig_24h_sub))),]
clue.erastin_24h_5um.sig_24h_sub <- clue.erastin_24h_5um.sig_24h_sub[order(match(rownames(clue.erastin_24h_5um.sig_24h_sub),rownames(sig_24h_sub))),]


set.seed(11)
Heatmap(rna.sig_24h_sub[,1:3],
        show_column_dend = F,
        show_row_dend = F,
        show_row_names = F,
        column_names_gp = gpar(fontsize = 7),
        column_title = 'RNA-seq',
        column_title_gp = gpar(fontsize = 8),
        show_heatmap_legend = F,
        na_col = 'black',
        cluster_rows = T,
        cluster_columns = T,
        row_split = 2) +
  
Heatmap(clue.erastin_6h_5um.sig_24h_sub,
        show_column_dend = F,
        show_row_dend = F,
        show_row_names = F,
        row_names_gp = gpar(fontsize = 5),
        column_names_gp = gpar(fontsize = 7),
        column_title = 'Clue-Erastin 5um-6h',
        column_title_gp = gpar(fontsize = 8),
        show_heatmap_legend = F,
        clustering_method_columns = 'complete') +
  
Heatmap(clue.erastin_24h_5um.sig_24h_sub,
        show_column_dend = F,
        show_row_dend = F,
        show_row_names = F,
        column_names_gp = gpar(fontsize = 7),
        column_title = '5um-24h',
        column_title_gp = gpar(fontsize = 8),
        show_heatmap_legend = F) +
  
  Heatmap(clue.erastin_24h_10um.sig_24h_sub,
        show_column_dend = F,
        show_row_dend = F,
        show_row_names = T,
        column_names_gp = gpar(fontsize = 7),
        row_names_gp = gpar(fontsize = 5),
        column_title = '10um-24h',
        column_title_gp = gpar(fontsize = 8),
        show_heatmap_legend = F)


```

## See how the gene signature derived from the 24h clue samples captures genes reported in elife
306 gene signature captures 15/37 genes
63 gene signature captures 6/37
```{r}
topGenes <- c('CHAC1', 'DDIT4', 'LOC284561', 'ASNS', 'TSC22D3', 'DDIT3', 'JDP2', 'SESN2',
              'SLC1A4', 'PCK2', 'SLC7A11', 'TXNIP', 'VLDLR', 'GPT2', 'PSAT1', 'C9ORF150',
              'SLC7A5', 'HERPUD1', 'XBP1', 'ATF3', 'SLC3A2', 'CBS', 'ATF4', 'ZNF419', 
              'KLHL24', 'TRIB3', 'ZNF643', 'ATP6V1G2', 'VEGFA', 'GDF15', 'TUBE1', 
              'ARRDC3', 'CEBPG', 'SNORA16A', 'RGS4', 'MUTED-TXNDC5', 'LOC390705')

table(topGenes %in% clue.erastin$pr_gene_symbol)
table(topGenes %in% rownames(sig_24h))
table(topGenes %in% rownames(sig_24h_sub))

```

## See how the gene signature derived from the 24h clue samples captures genes reported in crappy lit-pulled gene sig papers
306 gene signature captures 9/60 genes reported in lit-pulled 60 gene signature. 57/60 measured in clue datasest
306 gene signature captures 10/149 genes reported in lit-pulled 60 gene signature. 130/149 measured in clue datasest
60 lit-pulled gene signature captures 3/37 genes reported in elife gene signature
149 lit-pulled gene signature captures 7/37 genes reported in elife gene signature
149 lit-pulled gene signature captures 48/60 genes reported in 60 lit pulled gene signature

```{r}

# Taken from PMID: 32760210
# A Novel Ferroptosis-related Gene Signature for Overall Survival Prediction in Patients with Hepatocellular Carcinoma (Liang et al 2020)
# Basically same signature used for 3 other papers; 60 genes
litGenes60 <- readxl::read_xls('/mctp/share/users/rebernrj/projects/ferroptosisGeneSig/data/pmid_ 32760210/Table S1.xls')
litGenes60 <- litGenes60$`Ferrotosis-related genes`

# genes in clue dataset
table(litGenes60 %in% clue.erastin$pr_gene_symbol)
# genes in 306gs
table(litGenes60 %in% rownames(sig_24h))
# genes in 63gs
table(litGenes60 %in% rownames(sig_24h_sub))

# genes in elife gs
table(topGenes %in% litGenes60)


# Taken from DOI:https://doi.org/10.1016/j.omto.2021.02.011
# Systematic profiling of ferroptosis gene signatures predicts prognostic factors in esophageal squamous cell carcinoma (Lu et al 2021)
litGenes149 <- readxl::read_xlsx('/mctp/share/users/rebernrj/projects/ferroptosisGeneSig/data/doi_10.1016_j.omto.2021.02.11/mmc2.xlsx', skip = 1)
litGenes149 <- litGenes149$Symbol

# genes in clue dataset
table(litGenes149 %in% clue.erastin$pr_gene_symbol)
# genes in 306gs
table(litGenes149 %in% rownames(sig_24h))
# genes in 63gs
table(litGenes149 %in% rownames(sig_24h_sub))

# genes in elife gs
table(topGenes %in% litGenes149)

# genes in 60 lit pulled gs
table(litGenes60 %in% litGenes149)


```

## Create venn diagram of genes in common between signatures
```{r}

data = list(`306gs` = rownames(sig_24h), `Lit-pulled gs` = litGenes60, `elife gs` = topGenes)

ggvenn(
  data= data, 
  fill_color = c("#0073C2FF", "#EFC000FF", "#868686FF"),
  stroke_size = 0.5, set_name_size = 8,
  show_percentage = F,
  text_size = 8
  )


```


## save gene signature
```{r}
temp <- sig_24h 
colnames(temp)[4:16] <- paste0(colnames(sig_24h)[4:16], '_24h')
write.table(temp, '/mctp/share/users/rebernrj/projects/ferroptosisGeneSig/out/gene_signatures/erastin_clue24h_gse104462_306genes.txt',
            row.names = T, col.names = T, sep = '\t', quote = F)

temp <- sig_24h_sub
colnames(temp)[4:16] <- paste0(colnames(sig_24h)[4:16], '_24h')
write.table(temp, '/mctp/share/users/rebernrj/projects/ferroptosisGeneSig/out/gene_signatures/erastin_clue24h_gse104462_63genes.txt',
            row.names = T, col.names = T, sep = '\t', quote = F)

```



# Look specifically at 978 landmark genes from Clue dataset rather than entirety of imputed data
## Again start with the 24h samples 

## Get L1000 genes
```{r}

# Load metadata file
l1000 <- fread('/mctp/share/users/rebernrj/projects/ferroptosisGeneSig/data/clue/LINCS_L1000_metadata/GSE92742_Broad_LINCS_gene_info.txt', data.table = F)

# filter for L1000 genes
l1000 %<>% dplyr::filter(pr_is_lm == 1)

```

## Find the genes significant in the 24 hour data set (clue)
Will start with all of the clue genes (including imputed).
To do this:
1) merge both 24h clue datasets together (5um/10um)
2) Filter for genes in L1000 genes
3) Select genes which are changing the same direction in at least 10/13 samples
4) Filter the remaining genes to have an abs(sum of the z scores for the gene) > mean(abs(sum of the z scores for the gene))
```{r}

# Reload the 24 hour clue data
## 10um, 24h
clue.erastin_24h_10um <- clean_clue(clue.erastin, dose = '10 µM', time = '24 h', see_avail_opts = F)
## 5um, 24h
clue.erastin_24h_5um <- clean_clue(clue.erastin, dose = '5 µM', time = '24 h', see_avail_opts = F)

# merge data frames adding dosage info to colnames
## 10um
clue.erastin_24h_10um %<>% set_rownames(clue.erastin_24h_10um$pr_gene_symbol) %>%
  dplyr::select(-pr_gene_symbol)
colnames(clue.erastin_24h_10um) <- paste0(colnames(clue.erastin_24h_10um), '_10um')
## 5um
clue.erastin_24h_5um %<>% set_rownames(clue.erastin_24h_5um$pr_gene_symbol) %>%
  dplyr::select(-pr_gene_symbol)
colnames(clue.erastin_24h_5um) <- paste0(colnames(clue.erastin_24h_5um), '_5um')
## merge
clue.l1000 <- merge(clue.erastin_24h_5um, clue.erastin_24h_10um, by = 0)
clue.l1000 %<>% set_rownames(clue.l1000$Row.names) %>%
  dplyr::select(-Row.names)


# Select L1000 genes
clue.l1000 %<>% dplyr::filter(rownames(.) %in% l1000$pr_gene_symbol)


# Filter for those changing the same way in at least 10/13
clue.l1000 %<>% 
  dplyr::mutate(sums = (HA1E_5um > 0) + 
                  (A375_5um > 0) +
                  (HT29_5um > 0) +
                  (MCF7_5um > 0) +
                  (PC3_5um > 0) +
                  (A549_5um > 0) +
                  (HCC515_5um > 0) +
                  (VCAP_5um > 0) +
                  (HEPG2_10um > 0) +
                  (A375_10um > 0) +
                  (HCC515_10um > 0) +
                  (A549_10um > 0) +
                  (HA1E_10um > 0)) %>%
  dplyr::mutate(upregulated = ifelse(sums >= 8, T, F))
clue.l1000 %<>% dplyr::filter(sums >= 10 | sums <= 3) %>%
  dplyr::select(-sums)

# Find most significantly changing genes (summation of z scores)
clue.l1000 %<>%
  mutate(sums = rowSums(abs(clue.l1000[,-14])))
clue.l1000 %<>% dplyr::filter(sums >= mean(clue.l1000$sums))
clue.l1000 %<>% dplyr::select(-sums)


```


## see how many recapitulate in the good RNA-seq data set
```{r}
# GSE104462 (erastin)
gse104462 <- fread('/mctp/share/users/rebernrj/projects/ferroptosisGeneSig/int/GSE104462/deseq2/GSE104462_degs_erastin_control.txt', data.table = F)

gse104462 %<>%
  set_rownames(gse104462$Row.names) %>%
  dplyr::select(log2FoldChange, padj)

# Find genes present in both data sets
sig_l1000 <- merge(clue.l1000, gse104462, by = 0)

# Filter for those changing the same way in both
sig_l1000 %<>% dplyr::filter((upregulated == T & log2FoldChange > 0) | (upregulated == F & log2FoldChange < 0))

# filter for those significant in the RNA-seq data set
sig_l1000 %<>% dplyr::filter(padj < 0.1)

# clean for processing
sig_l1000 %<>% set_rownames(sig_l1000$Row.names) %>%
  dplyr::select(upregulated:padj, HA1E_5um:HA1E_10um)
colnames(sig_l1000)[2:3] <- c('GSE104462_LFC', 'GSE104462_padj')
sig_l1000 %<>% dplyr::arrange(GSE104462_LFC)


```

## Select sig_l1000 genes and order for heatmap
```{r}

# select genes
rna.sig_l1000 <- rna %>% dplyr::filter(rownames(.) %in% rownames(sig_l1000))
clue.erastin_6h_5um.sig_l1000 <- clue.erastin_6h_5um %>% dplyr::filter(rownames(.) %in% rownames(sig_l1000))
clue.erastin_24h_10um.sig_l1000 <- clue.erastin_24h_10um %>% dplyr::filter(rownames(.) %in% rownames(sig_l1000)) 
clue.erastin_24h_5um.sig_l1000 <- clue.erastin_24h_5um %>% dplyr::filter(rownames(.) %in% rownames(sig_l1000))

# order
rna.sig_l1000 <- rna.sig_l1000[order(match(rownames(rna.sig_l1000),rownames(sig_l1000))),]
clue.erastin_6h_5um.sig_l1000 <- clue.erastin_6h_5um.sig_l1000[order(match(rownames(clue.erastin_6h_5um.sig_l1000),rownames(sig_l1000))),]
clue.erastin_24h_10um.sig_l1000 <- clue.erastin_24h_10um.sig_l1000[order(match(rownames(clue.erastin_24h_10um.sig_l1000),rownames(sig_l1000))),]
clue.erastin_24h_5um.sig_l1000 <- clue.erastin_24h_5um.sig_l1000[order(match(rownames(clue.erastin_24h_5um.sig_l1000),rownames(sig_l1000))),]

```

## Plot gene signature in all data sets
```{r}

# row split
up <- table(rna.sig_l1000$GSE104462_LFC > 0)[2]
down <- table(rna.sig_l1000$GSE104462_LFC > 0)[1]

set.seed(11)
Heatmap(rna.sig_l1000[,1:3],
        show_column_dend = F,
        show_row_dend = F,
        show_row_names = F,
        column_names_gp = gpar(fontsize = 7),
        column_title = 'RNA-seq',
        column_title_gp = gpar(fontsize = 8),
        show_heatmap_legend = F,
        row_split = c(rep('down', down), rep('up', up)),
        na_col = 'black',
        cluster_rows = T) +
  
Heatmap(clue.erastin_6h_5um.sig_l1000,
        show_column_dend = F,
        show_row_dend = F,
        show_row_names = F,
        row_names_gp = gpar(fontsize = 5),
        column_names_gp = gpar(fontsize = 7),
        column_title = 'Clue-Erastin 5um-6h',
        column_title_gp = gpar(fontsize = 8),
        show_heatmap_legend = F,
        clustering_method_columns = 'complete') +
  
Heatmap(clue.erastin_24h_5um.sig_l1000,
        show_column_dend = F,
        show_row_dend = F,
        show_row_names = F,
        column_names_gp = gpar(fontsize = 7),
        column_title = '5um-24h',
        column_title_gp = gpar(fontsize = 8),
        show_heatmap_legend = F) +
  
  Heatmap(clue.erastin_24h_10um.sig_l1000,
        show_column_dend = F,
        show_row_dend = F,
        show_row_names = F,
        column_names_gp = gpar(fontsize = 7),
        column_title = '10um-24h',
        column_title_gp = gpar(fontsize = 8),
        show_heatmap_legend = F)

```

## Sort by 24h time points to try and pull out blocks of genes changing together 
```{r}

## order by 24h row sums
clue.erastin_24h_5um.sig_l1000 %<>% dplyr::mutate(sums = rowSums(abs(clue.erastin_24h_5um.sig_l1000))) 
## label as upreg/downreg
clue.erastin_24h_5um.sig_l1000$upregulated <- sig_l1000$upregulated
clue.erastin_24h_5um.sig_l1000 %<>%
  arrange(upregulated, -sums) %>%
  dplyr::select(-sums, -upregulated)

# order other datasets based on clue.erastin_6h_5um.sig_24h
rna.sig_l1000 <- rna.sig_l1000[order(match(rownames(rna.sig_l1000),rownames(clue.erastin_24h_5um.sig_l1000))),]
clue.erastin_6h_5um.sig_l1000 <- clue.erastin_6h_5um.sig_l1000[order(match(rownames(clue.erastin_6h_5um.sig_l1000),rownames(clue.erastin_24h_5um.sig_l1000))),]
clue.erastin_24h_10um.sig_l1000 <- clue.erastin_24h_10um.sig_l1000[order(match(rownames(clue.erastin_24h_10um.sig_l1000),rownames(clue.erastin_24h_5um.sig_l1000))),]




# row split
up <- table(rna.sig_l1000$GSE104462_LFC > 0)[2]
down <- table(rna.sig_l1000$GSE104462_LFC > 0)[1]

set.seed(11)
Heatmap(rna.sig_l1000[,1:3],
        show_column_dend = F,
        show_row_dend = F,
        show_row_names = F,
        column_names_gp = gpar(fontsize = 7),
        column_title = 'RNA-seq',
        column_title_gp = gpar(fontsize = 8),
        show_heatmap_legend = F,
        row_split = c(rep('down', down), rep('up', up)),
        na_col = 'black',
        cluster_rows = F,
        cluster_columns = T) +
  
Heatmap(clue.erastin_6h_5um.sig_l1000,
        show_column_dend = F,
        show_row_dend = F,
        show_row_names = F,
        row_names_gp = gpar(fontsize = 5),
        column_names_gp = gpar(fontsize = 7),
        column_title = 'Clue-Erastin 5um-6h',
        column_title_gp = gpar(fontsize = 8),
        show_heatmap_legend = F,
        clustering_method_columns = 'complete') +
  
Heatmap(clue.erastin_24h_5um.sig_l1000,
        show_column_dend = F,
        show_row_dend = F,
        show_row_names = F,
        column_names_gp = gpar(fontsize = 7),
        column_title = '5um-24h',
        column_title_gp = gpar(fontsize = 8),
        show_heatmap_legend = F) +
  
  Heatmap(clue.erastin_24h_10um.sig_l1000,
        show_column_dend = F,
        show_row_dend = F,
        show_row_names = T,
        column_names_gp = gpar(fontsize = 7),
        row_names_gp = gpar(fontsize = 5),
        column_title = '10um-24h',
        column_title_gp = gpar(fontsize = 8),
        show_heatmap_legend = F)

```

## Subset sig_l1000 further based on subclustering
#### Combine into single df and to improve clustering and subset of interesting genes
Does not subcluster into any further interesting clusters 
```{r}

# change colnames 
colnames(clue.erastin_6h_5um.sig_l1000) <- paste0(colnames(clue.erastin_6h_5um.sig_l1000), '_5um_6h')
colnames(clue.erastin_24h_5um.sig_l1000) <- paste0(colnames(clue.erastin_24h_5um.sig_l1000), '_24h')
colnames(clue.erastin_24h_10um.sig_l1000) <- paste0(colnames(clue.erastin_24h_10um.sig_l1000), '_24h')

sig_l1000_merge <- merge(clue.erastin_6h_5um.sig_l1000, clue.erastin_24h_5um.sig_l1000, by = 0)
sig_l1000_merge %<>% set_rownames(sig_l1000_merge$Row.names) %>% dplyr::select(-Row.names)
sig_l1000_merge <- merge(sig_l1000_merge, clue.erastin_24h_10um.sig_l1000, by = 0)
sig_l1000_merge %<>% set_rownames(sig_l1000_merge$Row.names) %>% dplyr::select(-Row.names)


# Plot
set.seed(11)
Heatmap(sig_l1000_merge,
        show_column_dend = F,
        show_row_dend = F,
        show_row_names = T,
        column_names_gp = gpar(fontsize = 7),
        row_names_gp = gpar(fontsize = 5),
        column_title = 'Combined Clue data',
        column_title_gp = gpar(fontsize = 8),
        show_heatmap_legend = F,
        row_km = 4,
        column_km = 2,
        row_km_repeats = 100,
        na_col = 'black',
        cluster_rows = T,
        cluster_columns = T)

```

## save gene signature
```{r}
temp <- sig_l1000 
colnames(temp)[4:16] <- paste0(colnames(sig_24h)[4:16], '_24h')
write.table(temp, '/mctp/share/users/rebernrj/projects/ferroptosisGeneSig/out/gene_signatures/erastin_clue24h_gse104462_46genes.txt',row.names = T, col.names = T, sep = '\t', quote = F)

```



# Gene signature based on correlation of 2 best RNA-seq data sets (elife and GSE104462)

```{r}

# Remove NA values from RNA-seq dataset
rna.cor <- rna %>% 
  dplyr::filter(!is.na(GSE104462_LFC) & !is.na(elife_LFC))
## scatter
plot(rna.cor$GSE104462_LFC, rna.cor$elife_LFC)

#Remove values with small LFC
rna.cor %<>% dplyr::filter(abs(GSE104462_LFC) > 1 & abs(elife_LFC) > 1)
plot(rna.cor$GSE104462_LFC, rna.cor$elife_LFC)

# Keep only top right and bottom left quadrants
rna.cor %<>% dplyr::filter((GSE104462_LFC > 1 & elife_LFC > 1) | (GSE104462_LFC < 1 & elife_LFC < 1))
plot(rna.cor$GSE104462_LFC, rna.cor$elife_LFC)

# Arrange for heatmap
rna.cor %<>% dplyr::arrange(GSE104462_LFC)


```

## Select rna.cor genes and order for heatmap
```{r}

# select genes
rna.rna.cor <- rna %>% dplyr::filter(rownames(.) %in% rownames(rna.cor)) %>%
  dplyr::filter(rownames(.) %in% clue.erastin$pr_gene_symbol)
clue.erastin_6h_5um.rna.cor <- clue.erastin_6h_5um %>% dplyr::filter(rownames(.) %in% rownames(rna.cor))
clue.erastin_24h_10um.rna.cor <- clue.erastin_24h_10um %>% dplyr::filter(rownames(.) %in% rownames(rna.cor)) 
clue.erastin_24h_5um.rna.cor <- clue.erastin_24h_5um %>% dplyr::filter(rownames(.) %in% rownames(rna.cor))

# order
rna.rna.cor <- rna.rna.cor[order(match(rownames(rna.rna.cor),rownames(rna.cor))),]
clue.erastin_6h_5um.rna.cor <- clue.erastin_6h_5um.rna.cor[order(match(rownames(clue.erastin_6h_5um.rna.cor),rownames(rna.cor))),]
clue.erastin_24h_10um.rna.cor <- clue.erastin_24h_10um.rna.cor[order(match(rownames(clue.erastin_24h_10um.rna.cor),rownames(rna.cor))),]
clue.erastin_24h_5um.rna.cor <- clue.erastin_24h_5um.rna.cor[order(match(rownames(clue.erastin_24h_5um.rna.cor),rownames(rna.cor))),]

```


## Plot gene signature in all data sets
Ordered based on GSE104462 LogFC
```{r}

# row split
up <- table(rna.rna.cor$GSE104462_LFC > 0)[2]
down <- table(rna.rna.cor$GSE104462_LFC > 0)[1]

set.seed(11)
Heatmap(rna.rna.cor[,1:3],
        show_column_dend = F,
        show_row_dend = F,
        show_row_names = F,
        column_names_gp = gpar(fontsize = 7),
        column_title = 'RNA-seq',
        column_title_gp = gpar(fontsize = 8),
        show_heatmap_legend = F,
        row_split = c(rep('down', down), rep('up', up)),
        na_col = 'black',
        cluster_rows = F) +
  
Heatmap(clue.erastin_6h_5um.rna.cor,
        show_column_dend = F,
        show_row_dend = F,
        show_row_names = F,
        row_names_gp = gpar(fontsize = 5),
        column_names_gp = gpar(fontsize = 7),
        column_title = 'Clue-Erastin 5um-6h',
        column_title_gp = gpar(fontsize = 8),
        show_heatmap_legend = F,
        clustering_method_columns = 'complete') +
  
Heatmap(clue.erastin_24h_5um.rna.cor,
        show_column_dend = F,
        show_row_dend = F,
        show_row_names = F,
        column_names_gp = gpar(fontsize = 7),
        column_title = '5um-24h',
        column_title_gp = gpar(fontsize = 8),
        show_heatmap_legend = F) +
  
  Heatmap(clue.erastin_24h_10um.rna.cor,
        show_column_dend = F,
        show_row_dend = F,
        show_row_names = F,
        column_names_gp = gpar(fontsize = 7),
        column_title = '10um-24h',
        column_title_gp = gpar(fontsize = 8),
        show_heatmap_legend = F)

```

## See how captures genes reported in elife
Captures 21/37 genes
```{r}
topGenes <- c('CHAC1', 'DDIT4', 'LOC284561', 'ASNS', 'TSC22D3', 'DDIT3', 'JDP2', 'SESN2',
              'SLC1A4', 'PCK2', 'SLC7A11', 'TXNIP', 'VLDLR', 'GPT2', 'PSAT1', 'C9ORF150',
              'SLC7A5', 'HERPUD1', 'XBP1', 'ATF3', 'SLC3A2', 'CBS', 'ATF4', 'ZNF419', 
              'KLHL24', 'TRIB3', 'ZNF643', 'ATP6V1G2', 'VEGFA', 'GDF15', 'TUBE1', 
              'ARRDC3', 'CEBPG', 'SNORA16A', 'RGS4', 'MUTED-TXNDC5', 'LOC390705')

table(topGenes %in% rownames(rna.cor))

```

## save gene signature
```{r}
temp <- rna.rna.cor 
write.table(temp, '/mctp/share/users/rebernrj/projects/ferroptosisGeneSig/out/gene_signatures/erastin_elife_correlation_gse104462_96genes.txt',
            row.names = T, col.names = T, sep = '\t', quote = F)

```





# PCA plots of the clue data using the signature developed from the cells exposed to erastin for 24 hours

## 307 gene signature
#### combine all clue data sets for pca plotting
```{r}

# combine
## rename cols for sorting
colnames(clue.erastin_24h_10um.sig_24h) <- paste0(colnames(clue.erastin_24h_10um.sig_24h), '_24h')
colnames(clue.erastin_24h_5um.sig_24h) <- paste0(colnames(clue.erastin_24h_5um.sig_24h), '_24h')
colnames(clue.erastin_6h_5um.sig_24h) <- paste0(colnames(clue.erastin_6h_5um.sig_24h), '_5um_6h')
## merge
clue.erastin.pca <- merge(clue.erastin_24h_10um.sig_24h, clue.erastin_24h_5um.sig_24h, by = 0)
clue.erastin.pca %<>% set_rownames(clue.erastin.pca$Row.names) %>% dplyr::select(-Row.names)
clue.erastin.pca <- merge(clue.erastin.pca, clue.erastin_6h_5um.sig_24h, by = 0)
clue.erastin.pca %<>% set_rownames(clue.erastin.pca$Row.names) %>% dplyr::select(-Row.names)

# Transpose for plotting and add group info
clue.erastin.pca %<>% t() %>% as.data.frame()
clue.erastin.pca %<>% mutate(group = sub('[A-Z0-9]{2,8}\\_', replacement = '', rownames(clue.erastin.pca)))

```

#### plot pca of all clue datasets for 306 gene signature
All cell lines included
This one has lots of variation among PC1 attributable to those cell lines with high z scores
```{r}

set.seed(11)
dat <- clue.erastin.pca
autoplot(dat %>% dplyr::select(-group) %>% prcomp(),
         data = dat,
         colour = 'group') +
  theme_cowplot(12)

```

#### Remove outlier cell lines
Eliminates the cells with excessively strong Z scores (tight controls?) 
```{r}

# eliminating values based on heatmap above
elim <- c('T3M10_5um_6h', 'CORL23_5um_6h', 'SNU1040_5um_6h', 'CL34_5um_6h', 'NCIH1694_5um_6h')

dat <- clue.erastin.pca %>% filter(!rownames(.) %in% elim)
# PC1/2
set.seed(11)
autoplot(dat %>% dplyr::select(-group) %>% prcomp(),
         data = dat,
         colour = 'group',
         x=1,
         y=2) +
  theme_cowplot(12)
# PC 2/3
set.seed(11)
autoplot(dat %>% dplyr::select(-group) %>% prcomp(),
         data = dat,
         colour = 'group',
         x=2,
         y=3) +
  theme_cowplot(12)


```

## L1000 117-gene signature
#### combine all clue data sets for pca plotting
```{r}

# combine
## rename cols for sorting
colnames(clue.erastin_24h_10um.sig_l1000) <- paste0(colnames(clue.erastin_24h_10um.sig_l1000), '_24h')
colnames(clue.erastin_24h_5um.sig_l1000) <- paste0(colnames(clue.erastin_24h_5um.sig_l1000), '_24h')
colnames(clue.erastin_6h_5um.sig_l1000) <- paste0(colnames(clue.erastin_6h_5um.sig_l1000), '_5um_6h')
## merge
clue.erastin.pca <- merge(clue.erastin_24h_10um.sig_l1000, clue.erastin_24h_5um.sig_l1000, by = 0)
clue.erastin.pca %<>% set_rownames(clue.erastin.pca$Row.names) %>% dplyr::select(-Row.names)
clue.erastin.pca <- merge(clue.erastin.pca, clue.erastin_6h_5um.sig_l1000, by = 0)
clue.erastin.pca %<>% set_rownames(clue.erastin.pca$Row.names) %>% dplyr::select(-Row.names)

# Transpose for plotting and add group info
clue.erastin.pca %<>% t() %>% as.data.frame()
clue.erastin.pca %<>% mutate(group = sub('[A-Z0-9]{2,8}\\_', replacement = '', rownames(clue.erastin.pca)))

```

#### All cell lines
```{r}

set.seed(11)
dat <- clue.erastin.pca
autoplot(dat %>% dplyr::select(-group) %>% prcomp(),
         data = dat,
         colour = 'group') +
  theme_cowplot(12)


```
#### Elminate outlier cell lines
```{r}
# eliminating values based on heatmap above
elim <- c('T3M10_5um_6h', 'CORL23_5um_6h', 'SNU1040_5um_6h', 'CL34_5um_6h', 'NCIH1694_5um_6h')

dat <- clue.erastin.pca %>% filter(!rownames(.) %in% elim)
# PC1/2
set.seed(11)
autoplot(dat %>% dplyr::select(-group) %>% prcomp(),
         data = dat,
         colour = 'group',
         x=1,
         y=2) +
  theme_cowplot(12)
```




# GO analysis of gene signatures


## Prepare resources for GO analysis 

### Prepare background list and get entrez gene ids
```{r}

# BiocManager::install("GOstats", dependencies = TRUE, INSTALL_opts = '--no-lock')
library('GOstats')
#BiocManager::install("biomaRt")
library("biomaRt")
#BiocManager::install("org.Hs.eg.db")
library('org.Hs.eg.db')

# Convert common names to entrez gene IDs
ensembl <- useMart("ensembl", dataset="hsapiens_gene_ensembl")
key <- getBM(attributes=c('external_gene_name', 'entrezgene_id'),
                    filters = 'chromosome_name',
                    values = c(1:22, 'X', 'Y'),
                    mart = ensembl,
                    useCache = F)

# get gene universe
universe <- merge(clue.erastin, key, by.x = 'pr_gene_symbol', by.y='external_gene_name')
universe <- unique(unlist(universe$entrezgene_id))


```

### Function to get and plot GO analysis
```{r}

go_analysis <- function(genes, universe, pcut = 1e-5, sig_name = 'XX gene signature'){
  
  ##################
  # Over-represented
  ##################
  
  # parameters for GO analysis
  params <- new('GOHyperGParams',
              geneIds = genes,
              universeGeneIds = universe,
              ontology = 'BP',
              pvalueCutoff = pcut,
              conditional = FALSE,
              testDirection = 'over',
              annotation = "org.Hs.eg.db"
             )
  
  # Run GO analysis
  over <- hyperGTest(params)
  
  # get -log10 pvalues for plotting 
  over <- summary(over)
  over$negLogP<- -1*log10(over$Pvalue)
  
  # factor Term so orders by pvalue
  over %<>% mutate(Term = factor(Term, levels = over$Term))
  
  # Over plot
  over.plot <- ggplot(over, aes(x = Term, y = negLogP)) +
    geom_col(fill = 'red') +
    theme_cowplot(12) +
    coord_flip() + 
    ggtitle(paste0('Over-enrichment: ', sig_name)) +
    xlab(NULL) + 
    ylab("-log10(pval)")
  
  ##################
  # Under-represented
  ##################
  
  params <- new('GOHyperGParams',
              geneIds = genes,
              universeGeneIds = universe,
              ontology = 'BP',
              pvalueCutoff = pcut,
              conditional = FALSE,
              testDirection = 'under',
              annotation = "org.Hs.eg.db"
             )
  
  # Run GO analysis
  under <- hyperGTest(params)
  
  # get -log10 pvalues for plotting 
  under <- summary(under)
  under$negLogP<- -1*log10(under$Pvalue)
  
  # factor Term so orders by pvalue
  under %<>% mutate(Term = factor(Term, levels = under$Term))
  
  # Under plot
  under.plot <- ggplot(under, aes(x = Term, y = negLogP)) +
    geom_col(fill = 'steelblue') +
    theme_cowplot(12) +
    coord_flip() +
    ggtitle(paste0('Under-enrichment: ', sig_name)) +
    xlab(NULL) + 
    ylab("-log10(pval)")
  
  return(list(over.plot, under.plot))
  
  
}

```

## Conduct GO analyses

### GO analysis of 24 hour signature (306 genes)
292 of 306 genes able to be converted to entrez
```{r, fig.height= 3, fig.width=6}

# get entrez ids for sig_24h
temp <- merge(sig_24h, key, by.x = 0, by.y='external_gene_name')

# perform GO enrichment
go_analysis(genes = temp$entrezgene_id, universe = universe, sig_name = '306 gene')

```

### GO analysis of 24 hour signature subset (63 genes)
59 of 63 genes able to be converted to entrez
```{r, fig.height= 3, fig.width=6}

# get entrez ids for sig_24h
temp <- merge(sig_24h_sub, key, by.x = 0, by.y='external_gene_name')
print(length(temp$Row.names))

# perform GO enrichment
go_analysis(genes = temp$entrezgene_id, universe = universe, sig_name = '63 gene')

```

### GO analysis of 24 hour signature subset of only the L1000 genes (46 genes)
44 of 46 genes able to be converted to entrez
```{r, fig.height= 3, fig.width=6}

# get entrez ids for sig_24h
temp <- merge(sig_l1000, key, by.x = 0, by.y='external_gene_name')
print(length(temp$Row.names))

# perform GO enrichment
go_analysis(genes = temp$entrezgene_id, universe = universe, sig_name = '43 gene')

```

### GO analysis of RNA-seq based signature obtained by correlation (96 genes)
96 of 96 genes able to be converted to entrez
```{r, fig.height= 3, fig.width=6}

# get entrez ids for sig_24h
temp <- merge(rna.rna.cor, key, by.x = 0, by.y='external_gene_name')
print(length(temp$Row.names))

# perform GO enrichment
go_analysis(genes = temp$entrezgene_id, universe = universe, sig_name = '96 gene')

```
















# Examine how elife, and other alternative gene signatures look within clue data

## Elife gene signature
### Select the elife genes and order for the heatmap
```{r}

topGenes <- c('CHAC1', 'DDIT4', 'LOC284561', 'ASNS', 'TSC22D3', 'DDIT3', 'JDP2', 'SESN2',
              'SLC1A4', 'PCK2', 'SLC7A11', 'TXNIP', 'VLDLR', 'GPT2', 'PSAT1', 'C9ORF150',
              'SLC7A5', 'HERPUD1', 'XBP1', 'ATF3', 'SLC3A2', 'CBS', 'ATF4', 'ZNF419', 
              'KLHL24', 'TRIB3', 'ZNF643', 'ATP6V1G2', 'VEGFA', 'GDF15', 'TUBE1', 
              'ARRDC3', 'CEBPG', 'SNORA16A', 'RGS4', 'MUTED-TXNDC5', 'LOC390705')

# select genes
rna.elife <- rna %>% dplyr::filter(rownames(.) %in% topGenes)
clue.erastin_6h_5um.elife <- clue.erastin_6h_5um %>% dplyr::filter(rownames(.) %in% topGenes)
clue.erastin_24h_10um.elife <- clue.erastin_24h_10um %>% dplyr::filter(rownames(.) %in% topGenes) 
clue.erastin_24h_5um.elife <- clue.erastin_24h_5um %>% dplyr::filter(rownames(.) %in% topGenes)

# Filter for genes present in clue dataset
rna.elife %<>% dplyr::filter(rownames(.) %in% clue.erastin$pr_gene_symbol)

# order
#rna.elife <- rna.elife[order(match(rownames(rna.elife),rownames(elife))),]
rna.elife %<>% dplyr::arrange(GSE104462_LFC)
clue.erastin_6h_5um.elife <- clue.erastin_6h_5um.elife[order(match(rownames(clue.erastin_6h_5um.elife),rownames(rna.elife))),]
clue.erastin_24h_10um.elife <- clue.erastin_24h_10um.elife[order(match(rownames(clue.erastin_24h_10um.elife),rownames(rna.elife))),]
clue.erastin_24h_5um.elife <- clue.erastin_24h_5um.elife[order(match(rownames(clue.erastin_24h_5um.elife),rownames(rna.elife))),]



```

### Plot elife gene signature in all data sets
Row ordering is based on clustering algorithm
```{r}

# row split
up <- table(rna.elife$GSE104462_LFC > 0)[2]
down <- table(rna.elife$GSE104462_LFC > 0)[1]

set.seed(11)
Heatmap(rna.elife[,1:3],
        show_column_dend = F,
        show_row_dend = F,
        show_row_names = F,
        column_names_gp = gpar(fontsize = 7),
        column_title = 'RNA-seq',
        column_title_gp = gpar(fontsize = 8),
        show_heatmap_legend = F,
        row_split = c(rep('down', down), rep('up', up)),
        na_col = 'black',
        cluster_rows = T) +
  
Heatmap(clue.erastin_6h_5um.elife,
        show_column_dend = F,
        show_row_dend = F,
        show_row_names = F,
        row_names_gp = gpar(fontsize = 5),
        column_names_gp = gpar(fontsize = 7),
        column_title = 'Clue-Erastin 5um-6h',
        column_title_gp = gpar(fontsize = 8),
        show_heatmap_legend = F,
        clustering_method_columns = 'complete') +
  
Heatmap(clue.erastin_24h_5um.elife,
        show_column_dend = F,
        show_row_dend = F,
        show_row_names = F,
        column_names_gp = gpar(fontsize = 7),
        column_title = '5um-24h',
        column_title_gp = gpar(fontsize = 8),
        show_heatmap_legend = F) +
  
  Heatmap(clue.erastin_24h_10um.elife,
        show_column_dend = F,
        show_row_dend = F,
        show_row_names = F,
        column_names_gp = gpar(fontsize = 7),
        column_title = '10um-24h',
        column_title_gp = gpar(fontsize = 8),
        show_heatmap_legend = F)

```


## 60 gene lit pulled ferroptosis gene signature
### Select the elife genes and order for the heatmap
```{r}

# Taken from PMID: 32760210
# A Novel Ferroptosis-related Gene Signature for Overall Survival Prediction in Patients with Hepatocellular Carcinoma (Liang et al 2020)
# Basically same signature used for 3 other papers; 60 genes
litGenes60 <- readxl::read_xls('/mctp/share/users/rebernrj/projects/ferroptosisGeneSig/data/pmid_ 32760210/Table S1.xls')
litGenes60 <- litGenes60$`Ferrotosis-related genes`


# select genes
rna.other60 <- rna %>% dplyr::filter(rownames(.) %in% litGenes60)
clue.erastin_6h_5um.other60 <- clue.erastin_6h_5um %>% dplyr::filter(rownames(.) %in% litGenes60)
clue.erastin_24h_10um.other60 <- clue.erastin_24h_10um %>% dplyr::filter(rownames(.) %in% litGenes60) 
clue.erastin_24h_5um.other60 <- clue.erastin_24h_5um %>% dplyr::filter(rownames(.) %in% litGenes60)

# Filter for genes present in clue dataset
rna.other60 %<>% dplyr::filter(rownames(.) %in% clue.erastin$pr_gene_symbol)

# order
#rna.other60 <- rna.other60[order(match(rownames(rna.other60),rownames(elife))),]
rna.other60 %<>% dplyr::arrange(GSE104462_LFC)
clue.erastin_6h_5um.other60 <- clue.erastin_6h_5um.other60[order(match(rownames(clue.erastin_6h_5um.other60),rownames(rna.other60))),]
clue.erastin_24h_10um.other60 <- clue.erastin_24h_10um.other60[order(match(rownames(clue.erastin_24h_10um.other60),rownames(rna.other60))),]
clue.erastin_24h_5um.other60 <- clue.erastin_24h_5um.other60[order(match(rownames(clue.erastin_24h_5um.other60),rownames(rna.other60))),]



```

### Plot other 60 lit pulled  gene signature in all data sets
Row ordering is based on clustering algorithm
```{r}

# row split
up <- table(rna.other60$GSE104462_LFC > 0)[2]
down <- table(rna.other60$GSE104462_LFC > 0)[1]

set.seed(11)
Heatmap(rna.other60[,1:3],
        show_column_dend = F,
        show_row_dend = F,
        show_row_names = F,
        column_names_gp = gpar(fontsize = 7),
        column_title = 'RNA-seq',
        column_title_gp = gpar(fontsize = 8),
        show_heatmap_legend = F,
        row_split = c(rep('down', down), rep('up', up)),
        na_col = 'black',
        cluster_rows = T) +
  
Heatmap(clue.erastin_6h_5um.other60,
        show_column_dend = F,
        show_row_dend = F,
        show_row_names = F,
        row_names_gp = gpar(fontsize = 5),
        column_names_gp = gpar(fontsize = 7),
        column_title = 'Clue-Erastin 5um-6h',
        column_title_gp = gpar(fontsize = 8),
        show_heatmap_legend = F,
        clustering_method_columns = 'complete') +
  
Heatmap(clue.erastin_24h_5um.other60,
        show_column_dend = F,
        show_row_dend = F,
        show_row_names = F,
        column_names_gp = gpar(fontsize = 7),
        column_title = '5um-24h',
        column_title_gp = gpar(fontsize = 8),
        show_heatmap_legend = F) +
  
  Heatmap(clue.erastin_24h_10um.other60,
        show_column_dend = F,
        show_row_dend = F,
        show_row_names = F,
        column_names_gp = gpar(fontsize = 7),
        column_title = '10um-24h',
        column_title_gp = gpar(fontsize = 8),
        show_heatmap_legend = F)

```





## 149 gene lit pulled ferroptosis gene signature
### Select the elife genes and order for the heatmap
```{r}

# Taken from DOI:https://doi.org/10.1016/j.omto.2021.02.011
# Systematic profiling of ferroptosis gene signatures predicts prognostic factors in esophageal squamous cell carcinoma (Lu et al 2021)
litGenes149 <- readxl::read_xlsx('/mctp/share/users/rebernrj/projects/ferroptosisGeneSig/data/doi_10.1016_j.omto.2021.02.11/mmc2.xlsx', skip = 1)
litGenes149 <- litGenes149$Symbol

# Remove IFNg and CDO1 because not present in any of the RNA-seq datasets
litGenes149 <- litGenes149[!litGenes149 %in% c('IFNG', 'CDO1')]

# select genes
rna.other149 <- rna %>% dplyr::filter(rownames(.) %in% litGenes149)
clue.erastin_6h_5um.other149 <- clue.erastin_6h_5um %>% dplyr::filter(rownames(.) %in% rownames(rna.other149))
clue.erastin_24h_10um.other149 <- clue.erastin_24h_10um %>% dplyr::filter(rownames(.) %in% rownames(rna.other149)) 
clue.erastin_24h_5um.other149 <- clue.erastin_24h_5um %>% dplyr::filter(rownames(.) %in% rownames(rna.other149))

# Filter for genes present in clue dataset
rna.other149 %<>% dplyr::filter(rownames(.) %in% clue.erastin$pr_gene_symbol)

# order
#rna.other149 <- rna.other149[order(match(rownames(rna.other149),rownames(elife))),]
rna.other149 %<>% dplyr::arrange(GSE104462_LFC)
clue.erastin_6h_5um.other149 <- clue.erastin_6h_5um.other149[order(match(rownames(clue.erastin_6h_5um.other149),rownames(rna.other149))),]
clue.erastin_24h_10um.other149 <- clue.erastin_24h_10um.other149[order(match(rownames(clue.erastin_24h_10um.other149),rownames(rna.other149))),]
clue.erastin_24h_5um.other149 <- clue.erastin_24h_5um.other149[order(match(rownames(clue.erastin_24h_5um.other149),rownames(rna.other149))),]



```

### Plot other 149 lit pulled gene signature in all data sets
Row ordering is based on clustering algorithm
```{r}

# row split
up <- table(rna.other149$GSE104462_LFC > 0)[2]
down <- table(rna.other149$GSE104462_LFC > 0)[1]


set.seed(11)
Heatmap(rna.other149[,1:3],
        show_column_dend = F,
        show_row_dend = F,
        show_row_names = F,
        column_names_gp = gpar(fontsize = 7),
        column_title = 'RNA-seq',
        column_title_gp = gpar(fontsize = 8),
        show_heatmap_legend = F,
        row_split = c(rep('down', down), rep('up', up)),
        na_col = 'black',
        cluster_rows = T) +
  
Heatmap(clue.erastin_6h_5um.other149,
        show_column_dend = F,
        show_row_dend = F,
        show_row_names = F,
        row_names_gp = gpar(fontsize = 5),
        column_names_gp = gpar(fontsize = 7),
        column_title = 'Clue-Erastin 5um-6h',
        column_title_gp = gpar(fontsize = 8),
        show_heatmap_legend = F,
        clustering_method_columns = 'complete') +
  
Heatmap(clue.erastin_24h_5um.other149,
        show_column_dend = F,
        show_row_dend = F,
        show_row_names = F,
        column_names_gp = gpar(fontsize = 7),
        column_title = '5um-24h',
        column_title_gp = gpar(fontsize = 8),
        show_heatmap_legend = F) +
  
  Heatmap(clue.erastin_24h_10um.other149,
        show_column_dend = F,
        show_row_dend = F,
        show_row_names = F,
        column_names_gp = gpar(fontsize = 7),
        column_title = '10um-24h',
        column_title_gp = gpar(fontsize = 8),
        show_heatmap_legend = F)

```








