---
title: "eda_nbl_cohort"
author: "Ryan Rebernick"
date: "4/4/2021"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, fig.keep = 'all')
```


# Description:
This script explores the NBL cohort by using the created ferroptosis gene signatures to explore scores in comparisons of interest. These include: 
- LR vs IR vs HR
- Dinutuximab response (Group 1-4)
- Pre/Post Dinutuximab
- Event-free survival
  - Split by mean
  - Split by quartile
- Overall survival
  - Split by mean
  - Split by quartile
Comparisons were done using 4 gene signatures created for ferroptosis. These gene signatures were scored using 3 methods: Nondirectional Z scores, Directional Z-scores, and SingScore (https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-018-2435-4). 


# Load packages
```{r, warning=F, message=FALSE}

library('plyr')
library('dplyr')
library('stringr')
library('data.table')
library('tidyverse')
library('magrittr')

#BiocManager::install("singscore")
library('singscore')
#install.packages("ggpubr")
library('ggpubr')
library('ggfortify')
library('cowplot')

```


# Load NBL cohort data and select samples for analysis
## Load NBL counts data and merge with metadata
17 duplicate gene names are present. These are just removed. If they are needed for any of the gene signatures they can be added to the dataset later. 
```{r}

# Load RNA-seq counts
nbl <- fread('/mctp/share/users/rebernrj/projects/ferroptosisGeneSig/data/nbl_cohort/release/ForRyan/nbl_expression-cts.csv', data.table = F)

# Remove any duplicate gene_ids so can transpose matrix
## View duplicate common gene_ids
duplicates <- nbl$gene_name[nbl$gene_name %>% duplicated()]
## remove duplicates
nbl %<>% dplyr::filter(!gene_name %in% duplicates)

# reformat the RNA-seq counts for merging with the clinical data
nbl %<>% set_rownames(nbl$gene_name) %>%
  dplyr::select(-gene_name, -gene_id) %>%
  t() %>%
  as.data.frame()

# Load the clinical data
temp <- fread('/mctp/share/users/rebernrj/projects/ferroptosisGeneSig/data/nbl_cohort/release/ForRyan/20210403_ForRyan_2021_0403_nbl_rnaseq.txt', data.table = F)

# Merge dataframes for downstream use
nbl <- merge(temp, nbl, by.x = 'rlib', by.y = 0)


```

## Select samples needed
Filtered for samples based on the following criteria:
- RNA capture method of 'capt'; ensured uniform RNA capture method
- Nonzero purity scores
- For rebiopsies of patient, took the sample with the higher purity w/in a timepoint (i.e. dxprim, dx2prim, ect)
- Eliminated samples which didn't fit the HR/LR/IR criteria (1 sample)
```{r}

# Filter for desired samples
nbl %<>% 
  ## Filter for the 'capt' RNA capture method
  dplyr::filter(r.capture == 'capt') %>%
  ## filter for nonzero purity of RNA
  dplyr::filter(purity > 0) %>%
  ## Combine timepoint refrel and relapse variables
  dplyr::mutate(timepoint = ifelse(timepoint %in% c('refrel', 'relapse'), 'rel_ref', timepoint)) %>%
  ## Generalize the timepoint variables
  dplyr::mutate(genTime = ifelse(timepoint %in% c('dxprim', 'dx2prim', 'dxmet'), 'dx', timepoint)) %>%
  ## Arrange by RNA purity for duplicate removal
  dplyr::arrange(timepoint, -purity) %>%
  ## Create pseudo variable for duplicate removal (ex. PO_3133)
  dplyr::mutate(pt.time = paste0(patient, '.', timepoint))
## show duplicates
nbl$patient[duplicated(nbl$pt.time)]
## remove duplicates based on purity, keeping highest purity
nbl <- nbl[!duplicated(nbl$pt.time),]
## Ensure samples represent only risk rated groups
nbl %<>% dplyr::filter(group %in% c('HR', 'IR', 'LR'))

# Create comparison variable for future use
nbl$comp <- ifelse(nbl$group %in% c('IR', "LR"), 'IR/LR', 'HR')

# Order dataframe cols
nbl %<>% dplyr::select(rlib:MYCN.cycle, comp, genTime, ARF5:AC106791.3)


```


# Create functions neeed to analyze gene signatures

## Create a function to return a signature scores using 3 methods:
1) Sample scoring package - less dependence of score on other samples
https://bioconductor.org/packages/devel/bioc/vignettes/singscore/inst/doc/singscore.html#21_Sample_scoring
2) Nondirectional Summation of Z scores
- Sum the absolute values of Z scores across up and downregulated genes
3) Directional Summation of Z scores
- Sum z scores of upregulated genes + -1*(Sum of Z scores of downregulated genes)
```{r}

# nbl_df - the cleaned nbl dataframe from above
# upGenes - list of genes upregulated in gene signature
# downGenes - list of genes downregulated in gene signature
# metacols - the columns of metadata within nbl_df

calculate_signature_scores <- function(nbl_df = nbl, upGenes, downGenes, metacols = 1:30){
  
  # create temp dataframe
  cnts <- nbl_df %>% set_rownames(paste0('Sample_', 1:length(nbl_df$rlib)))
  
  # return genes not in provided gene list and remove from query
  print(paste0('Up genes not in count matrix: ', upGenes[!upGenes %in% colnames(cnts)]))
  print(paste0('Down genes not in count matrix: ', downGenes[!downGenes %in% colnames(cnts)]))
  upGenes <- upGenes[upGenes %in% colnames(cnts)]
  downGenes <- downGenes[downGenes %in% colnames(cnts)]
  
  
  # Score samples using singScore and merge
  ## score
  rankData <- cnts %>% dplyr::select(-metacols) %>%  t() %>% rankGenes()
  scoredf <- simpleScore(rankData, upSet = upGenes, downSet = downGenes)
  colnames(scoredf) <- paste0('ss.', colnames(scoredf))
  ## merge SingScore Total score with tmp df
  cnts <- merge(cnts, scoredf, by = 0)
  
  
  # Calcluate nondirectional z scores 
  temp <- apply(cnts[,c(upGenes, downGenes)], 2, scale)
  temp$nondirectional_Zsum <- rowSums(abs(temp))
  ## Add Z score to dataframe 
  cnts$nondirectional_Zsum <- temp$nondirectional_Zsum
  
  
  # Calculate directional Z scores
  ## Upregulated genes
  temp <- apply(cnts %>% dplyr::select(all_of(upGenes)), 2, scale)
  temp$upZsum <- rowSums(temp)
  ## Add Z score to dataframe 
  cnts$upZsum <- temp$upZsum
  ## Downregulated genes
  temp <- apply(cnts %>% dplyr::select(all_of(downGenes)), 2, scale)
  temp$downZsum <- rowSums(temp)
  ## Add Z score to dataframe 
  cnts$downZsum <- temp$downZsum
  
  
  # Select columns
  cnts %<>% dplyr::select(metacols, ss.TotalScore:ss.DownDispersion, upZsum, downZsum)
  cnts %<>% mutate(directional_Zsum = upZsum + (downZsum * -1)) %>%
    dplyr::select(-Row.names)
  
  # Correlation
  print(paste0('Correlation b/t SingScore and Directional Zsum: ', cor(cnts$ss.TotalScore, cnts$directional_Zsum)))
  
  # Return
  return(cnts)
  
  
}

```

## Function to plot comparisons of interest
```{r}

# DF - dataframe to use; nbl format
# UPGENES - list of upregulated genes
# DOWNGENES - list of downregulated genes
# COMP - comparison variable/column
# STAT_TEST - c('wilcox.test', 't.test', 'anova', 'kruskal.test')
# TIMEPOINT = timepoint in nbl cohort to look at
make_plots <- function(
  DF,
  UPGENES, 
  DOWNGENES, 
  COMP='comp', 
  TIMEPOINT, 
  STAT_TEST = 'wilcox.test',
  TITLE = 'XX gene sig'){
  
  ####################
  # PCA PLOT by group
  ####################
  set.seed(11)
  dat <- DF[, c(COMP, UPGENES, DOWNGENES)]
  a <- autoplot(dat %>% dplyr::select(-COMP) %>% prcomp(),
           data = dat,
           colour = COMP,
           main = paste0('PCA using ', TITLE)) +
    theme_cowplot(12)
  
  
  ####################
  # Calclulate scores
  ####################
  scores <- calculate_signature_scores(nbl_df = DF, 
                                   upGenes = UPGENES,
                                   downGenes = DOWNGENES)
  
  ####################
  # Boxplot 
  ####################

  # select samples
  if(TIMEPOINT %in% c('dx2prim', 'dxmet', 'dxprim', 'posttx', 'rel_ref')){
    DF %<>% dplyr::filter(timepoint %in% c(TIMEPOINT))
  } else if(TIMEPOINT == 'all'){
    print('All timepoints included')
  } else {
    break
  }
  
  # Plot 
  b1 <- scores %>% ggplot(aes(x=get(COMP), y=ss.TotalScore)) + 
    geom_boxplot() +
    geom_jitter(alpha=0.6, width=0.15) +
    theme_bw(base_size=16) +
    labs(subtitle="SingScore by Risk group in dxprim") +
    stat_compare_means(method = STAT_TEST) +
    xlab(NULL)
  
  # Plot 
  # b2 <- scores %>% ggplot(aes(x=get(COMP), y=nondirectional_Zsum)) + 
  #   geom_boxplot() +
  #   geom_jitter(alpha=0.6, width=0.15) +
  #   theme_bw(base_size=16) +
  #   labs(subtitle="Nondirectional Z score by Risk group in dxprim") +
  #   stat_compare_means(method = STAT_TEST) +
  #   xlab(NULL)
  
  # Plot 
  b3 <- scores %>% ggplot(aes(x=get(COMP), y=directional_Zsum)) +
    geom_boxplot() +
    geom_jitter(alpha=0.6, width=0.15) +
    theme_bw(base_size=16) +
    labs(subtitle="Directional Z score by Risk group in dxprim") +
    stat_compare_means(method = STAT_TEST) +
    xlab(NULL)
  
  return(list(scores,a,b1,b3))
  
}

```


# Analyze gene signatures 

## 306 gene signature

### Load gene signature
```{r}
sig306 <- fread('/mctp/share/users/rebernrj/projects/ferroptosisGeneSig/out/gene_signatures/erastin_clue24h_gse104462_306genes.txt',
                data.table = F)

# Upregulated
sig306_up <- sig306 %>% 
  dplyr::filter(upregulated == T) %>% 
  dplyr::select(V1) %>% unlist()

# Downregulated
sig306_dn <- sig306 %>% 
  dplyr::filter(upregulated == F) %>% 
  dplyr::select(V1) %>% unlist()


```
### 306 gene signature for LR/IR/HR
```{r}

# Define comparison
nbl$comp <- nbl$group

# get PCA and boxplots
cur <- make_plots(
  DF = nbl,
  UPGENES = sig306_up, 
  DOWNGENES = sig306_dn, 
  COMP = 'comp', 
  TIMEPOINT = 'dxprim', 
  STAT_TEST = 'anova',
  TITLE = '306 gene sig')

# plot and get df
cur[2]
cur[3]
cur[4]
cur <- cur[[1]]

# select HR and group into 2 subgroups seen
cur %<>% dplyr::filter(comp == 'HR') %>%
  mutate(subset = ifelse(ss.TotalScore > -.05, 'sub_hi', 'sub_lo'))

# table vars of interest - proportions
prop.table(table(cur$subset, cur$timepoint), margin = 1) %>% `*`(100) %>% round(2) %>% barplot(beside=T, legend=rownames(.))
prop.table(table(cur$subset, cur$dinuresp_old), margin = 1) %>% `*`(100) %>% round(2) %>% barplot(beside=T, legend=rownames(.))
prop.table(table(cur$subset, cur$dinu_treat_prepost), margin = 1) %>% `*`(100) %>% round(2) %>% barplot(beside=T, legend=rownames(.))
prop.table(table(cur$subset, cur$site), margin = 1) %>% `*`(100) %>% round(2) %>% barplot(beside=T, legend=rownames(.))

# Raw values
table(cur$dinu_treat_prepost, cur$subset) %>% barplot(beside=T, legend=rownames(.))


```
### 306 gene signature for Dinutuxumab response among HR (1/2/3/4)
```{r}

# Define comparison
nbl$comp <- nbl$dinuresp_old
dat <- nbl %>% dplyr::filter(!comp %in% c('nothr', 'pending'))

# plot PCA and boxplots
make_plots(
  DF = dat,
  UPGENES = sig306_up, 
  DOWNGENES = sig306_dn, 
  COMP = 'comp', 
  TIMEPOINT = 'dxprim', 
  STAT_TEST = 'anova',
  TITLE = '306 gene sig')

```
### 306 gene signature for Dinutuxumab response among HR (1/234)
```{r}

# Define comparison
nbl$comp <- nbl$dinuresp_old
nbl %<>% mutate(comp = ifelse(comp %in% c('group2', 'group3', 'group4'), 'group234', comp))
dat <- nbl %>% dplyr::filter(!comp %in% c('pending'))

# plot PCA and boxplots
make_plots(
  DF = dat,
  UPGENES = sig306_up, 
  DOWNGENES = sig306_dn, 
  COMP = 'comp', 
  TIMEPOINT = 'dxprim', 
  STAT_TEST = 'anova',
  TITLE = '306 gene sig')

```
### 306 gene signature for Dinutuxumab response pre/post Dinituximab
```{r}

# Define comparison
nbl$comp <- nbl$dinu_treat_prepost
dat <- nbl %>% dplyr::mutate(comp = ifelse(!comp %in% c('postDIN', 'preDIN'), 'noDIN', comp))

# plot PCA and boxplots
make_plots(
  DF = dat,
  UPGENES = sig306_up, 
  DOWNGENES = sig306_dn, 
  COMP = 'comp', 
  TIMEPOINT = 'all', 
  STAT_TEST = 'anova',
  TITLE = '306 gene sig')

```
### 306 gene signature for Dinutuxumab response pre/post Dinituximab split by Dinitux_response
```{r}

# Define comparison
nbl$comp <- nbl$dinu_treat_prepost
dat <- nbl %>% dplyr::mutate(comp = ifelse(!comp %in% c('postDIN', 'preDIN'), 'noDIN', comp))

# calculate scores
scores <- calculate_signature_scores(nbl_df = dat, 
                                     upGenes = sig306_up,
                                     downGenes = sig306_dn)

# Plot 
scores %<>% dplyr::filter(dinuresp_old != 'pending') %>%
  mutate(dinuresp_old = ifelse(dinuresp_old %in% c('group2', 'group3', 'group4'), 'group234', dinuresp_old))
scores %>% ggplot(., aes(x=comp, y=ss.TotalScore, color=dinuresp_old)) + 
  geom_boxplot(color='black') +
  geom_jitter(alpha=0.6, width=0.15) +
  theme_bw(base_size=16) +
  labs(subtitle="SingScore: Biopsy time with response group") +
  xlab(NULL)

```
### 306 gene signature for Event Free Survival ( split by mean)
```{r}

# Quantiles of EFS time
q <- quantile(nbl$dinu_efstime, na.rm = T)

# Define comparison
nbl$comp <- ifelse(nbl$dinu_efstime >= q[3], 'Long EFS', 'Short EFS')
dat <- nbl %>% dplyr::filter(!is.na(dinu_efstime))


# plot PCA and boxplots
make_plots(
  DF = dat,
  UPGENES = sig306_up, 
  DOWNGENES = sig306_dn, 
  COMP = 'comp', 
  TIMEPOINT = 'all', 
  STAT_TEST = 't.test',
  TITLE = '306 gene sig')

```
### 306 gene signature for Event Free Survival ( split by quartile )
```{r}

# get quartiles
q <- quantile(nbl$dinu_efstime, na.rm = T)

# Define comparison
nbl$comp <- ifelse(nbl$dinu_efstime >= q[3], '3', '2')
nbl$comp <- ifelse(nbl$dinu_efstime >= q[4], '4', nbl$comp)
nbl$comp <- ifelse(nbl$dinu_efstime < q[2], '1', nbl$comp)
dat <- nbl %>% dplyr::filter(!is.na(dinu_efstime))


# plot PCA and boxplots
make_plots(
  DF = dat,
  UPGENES = sig306_up, 
  DOWNGENES = sig306_dn, 
  COMP = 'comp', 
  TIMEPOINT = 'all', 
  STAT_TEST = 'anova',
  TITLE = '306 gene sig')

```
### 306 gene signature for Overall Survival ( split by mean)
```{r}

# Quantiles of EFS time
q <- quantile(nbl$dinu_stime, na.rm = T)

# Define comparison
nbl$comp <- ifelse(nbl$dinu_stime >= q[3], 'Long OS', 'Short OS')
dat <- nbl %>% dplyr::filter(!is.na(dinu_stime))


# plot PCA and boxplots
make_plots(
  DF = dat,
  UPGENES = sig306_up, 
  DOWNGENES = sig306_dn, 
  COMP = 'comp', 
  TIMEPOINT = 'all', 
  STAT_TEST = 't.test',
  TITLE = '306 gene sig')

```
### 306 gene signature for Overall Survival ( split by quartile )
```{r}

# get quartiles
q <- quantile(nbl$dinu_stime, na.rm = T)

# Define comparison
nbl$comp <- ifelse(nbl$dinu_stime >= q[3], '3', '2')
nbl$comp <- ifelse(nbl$dinu_stime >= q[4], '4', nbl$comp)
nbl$comp <- ifelse(nbl$dinu_stime < q[2], '1', nbl$comp)
dat <- nbl %>% dplyr::filter(!is.na(dinu_stime))


# plot PCA and boxplots
make_plots(
  DF = dat,
  UPGENES = sig306_up, 
  DOWNGENES = sig306_dn, 
  COMP = 'comp', 
  TIMEPOINT = 'all', 
  STAT_TEST = 'anova',
  TITLE = '306 gene sig')

```
### 306 gene signature for HR (1/234) with additional info on pre/post DIN
```{r}

# Define comparison
nbl$comp <- nbl$dinuresp_old
nbl %<>% mutate(comp = ifelse(comp %in% c('group2', 'group3', 'group4'), 'group234', comp))
dat <- nbl %>% dplyr::filter(!comp %in% c('pending'))

# calculate scores
scores <- calculate_signature_scores(nbl_df = dat, 
                                     upGenes = sig306_up,
                                     downGenes = sig306_dn)

# Add 
scores %<>% mutate(subset = ifelse(ss.TotalScore > -.05, 'sub_hi', 'sub_lo')) %>%
  mutate(dinu_treat_prepost = ifelse(dinu_treat_prepost %in% c('postDIN', 'preDIN'), dinu_treat_prepost, 'unknown'))

# Plot 
# scores %<>% dplyr::filter(dinutx == 1)
scores %>% ggplot(., aes(x=comp, y=ss.TotalScore, color=dinu_treat_prepost)) + 
  geom_boxplot(color='black') +
  geom_jitter(alpha=0.6, width=0.15) +
  theme_bw(base_size=16) +
  labs(subtitle="SingScore by Risk group in dxprim") +
  xlab(NULL)

```








## 63 gene signature

### Load gene signature
```{r}
sig63 <- fread('/mctp/share/users/rebernrj/projects/ferroptosisGeneSig/out/gene_signatures/erastin_clue24h_gse104462_63genes.txt',
                data.table = F)

# Upregulated
sig63_up <- sig63 %>% 
  dplyr::filter(upregulated == T) %>% 
  dplyr::select(V1) %>% unlist()

# Downregulated
sig63_dn <- sig63 %>% 
  dplyr::filter(upregulated == F) %>% 
  dplyr::select(V1) %>% unlist()


```
### 63 gene signature for LR/IR/HR
```{r}

# Define comparison
nbl$comp <- nbl$group

# plot PCA and boxplots
make_plots(
  DF = nbl,
  UPGENES = sig63_up, 
  DOWNGENES = sig63_dn, 
  COMP = 'comp', 
  TIMEPOINT = 'dxprim', 
  STAT_TEST = 'anova',
  TITLE = '63 gene sig')

```
### 63 gene signature for Dinutuxumab response among HR
```{r}

# Define comparison
nbl$comp <- nbl$dinuresp_old
dat <- nbl %>% dplyr::filter(!comp %in% c('nothr', 'pending'))

# plot PCA and boxplots
make_plots(
  DF = dat,
  UPGENES = sig63_up, 
  DOWNGENES = sig63_dn, 
  COMP = 'comp', 
  TIMEPOINT = 'dxprim', 
  STAT_TEST = 'anova',
  TITLE = '63 gene sig')

```
### 63 gene signature for Dinutuxumab response pre/post Dinituximab
```{r}

# Define comparison
nbl$comp <- nbl$dinu_treat_prepost
dat <- nbl %>% dplyr::filter(comp %in% c('postDIN', 'preDIN'))

# plot PCA and boxplots
make_plots(
  DF = dat,
  UPGENES = sig63_up, 
  DOWNGENES = sig63_dn, 
  COMP = 'comp', 
  TIMEPOINT = 'all', 
  STAT_TEST = 't.test',
  TITLE = '63 gene sig')

```
### 63 gene signature for Event Free Survival ( split by mean)
```{r}

# Quantiles of EFS time
q <- quantile(nbl$dinu_efstime, na.rm = T)

# Define comparison
nbl$comp <- ifelse(nbl$dinu_efstime >= q[3], 'Long EFS', 'Short EFS')
dat <- nbl %>% dplyr::filter(!is.na(dinu_efstime))


# plot PCA and boxplots
make_plots(
  DF = dat,
  UPGENES = sig63_up, 
  DOWNGENES = sig63_dn, 
  COMP = 'comp', 
  TIMEPOINT = 'all', 
  STAT_TEST = 't.test',
  TITLE = '63 gene sig')

```
### 63 gene signature for Event Free Survival ( split by quartile )
```{r}

# get quartiles
q <- quantile(nbl$dinu_efstime, na.rm = T)

# Define comparison
nbl$comp <- ifelse(nbl$dinu_efstime >= q[3], '3', '2')
nbl$comp <- ifelse(nbl$dinu_efstime >= q[4], '4', nbl$comp)
nbl$comp <- ifelse(nbl$dinu_efstime < q[2], '1', nbl$comp)
dat <- nbl %>% dplyr::filter(!is.na(dinu_efstime))


# plot PCA and boxplots
make_plots(
  DF = dat,
  UPGENES = sig63_up, 
  DOWNGENES = sig63_dn, 
  COMP = 'comp', 
  TIMEPOINT = 'all', 
  STAT_TEST = 'anova',
  TITLE = '63 gene sig')

```
### 63 gene signature for Overall Survival ( split by mean)
```{r}

# Quantiles of EFS time
q <- quantile(nbl$dinu_stime, na.rm = T)

# Define comparison
nbl$comp <- ifelse(nbl$dinu_stime >= q[3], 'Long OS', 'Short OS')
dat <- nbl %>% dplyr::filter(!is.na(dinu_stime))


# plot PCA and boxplots
make_plots(
  DF = dat,
  UPGENES = sig63_up, 
  DOWNGENES = sig63_dn, 
  COMP = 'comp', 
  TIMEPOINT = 'all', 
  STAT_TEST = 't.test',
  TITLE = '63 gene sig')

```
### 63 gene signature for Overall Survival ( split by quartile )
```{r}

# get quartiles
q <- quantile(nbl$dinu_stime, na.rm = T)

# Define comparison
nbl$comp <- ifelse(nbl$dinu_stime >= q[3], '3', '2')
nbl$comp <- ifelse(nbl$dinu_stime >= q[4], '4', nbl$comp)
nbl$comp <- ifelse(nbl$dinu_stime < q[2], '1', nbl$comp)
dat <- nbl %>% dplyr::filter(!is.na(dinu_stime))


# plot PCA and boxplots
make_plots(
  DF = dat,
  UPGENES = sig63_up, 
  DOWNGENES = sig63_dn, 
  COMP = 'comp', 
  TIMEPOINT = 'all', 
  STAT_TEST = 'anova',
  TITLE = '63 gene sig')

```

## 46 gene signature

### Load gene signature
```{r}
sig46 <- fread('/mctp/share/users/rebernrj/projects/ferroptosisGeneSig/out/gene_signatures/erastin_clue24h_gse104462_46genes.txt',
                data.table = F)

# Upregulated
sig46_up <- sig46 %>% 
  dplyr::filter(upregulated == T) %>% 
  dplyr::select(V1) %>% unlist()

# Downregulated
sig46_dn <- sig46 %>% 
  dplyr::filter(upregulated == F) %>% 
  dplyr::select(V1) %>% unlist()


```
### 46 gene signature for LR/IR/HR
```{r}

# Define comparison
nbl$comp <- nbl$group

# plot PCA and boxplots
make_plots(
  DF = nbl,
  UPGENES = sig46_up, 
  DOWNGENES = sig46_dn, 
  COMP = 'comp', 
  TIMEPOINT = 'dxprim', 
  STAT_TEST = 'anova',
  TITLE = '46 gene sig')

```
### 46 gene signature for Dinutuxumab response among HR
```{r}

# Define comparison
nbl$comp <- nbl$dinuresp_old
dat <- nbl %>% dplyr::filter(!comp %in% c('nothr', 'pending'))

# plot PCA and boxplots
make_plots(
  DF = dat,
  UPGENES = sig46_up, 
  DOWNGENES = sig46_dn, 
  COMP = 'comp', 
  TIMEPOINT = 'dxprim', 
  STAT_TEST = 'anova',
  TITLE = '46 gene sig')

```
### 46 gene signature for Dinutuxumab response pre/post Dinituximab
```{r}

# Define comparison
nbl$comp <- nbl$dinu_treat_prepost
dat <- nbl %>% dplyr::filter(comp %in% c('postDIN', 'preDIN'))

# plot PCA and boxplots
make_plots(
  DF = dat,
  UPGENES = sig46_up, 
  DOWNGENES = sig46_dn, 
  COMP = 'comp', 
  TIMEPOINT = 'all', 
  STAT_TEST = 't.test',
  TITLE = '46 gene sig')

```
### 46 gene signature for Event Free Survival ( split by mean)
```{r}

# Quantiles of EFS time
q <- quantile(nbl$dinu_efstime, na.rm = T)

# Define comparison
nbl$comp <- ifelse(nbl$dinu_efstime >= q[3], 'Long EFS', 'Short EFS')
dat <- nbl %>% dplyr::filter(!is.na(dinu_efstime))


# plot PCA and boxplots
make_plots(
  DF = dat,
  UPGENES = sig46_up, 
  DOWNGENES = sig46_dn, 
  COMP = 'comp', 
  TIMEPOINT = 'all', 
  STAT_TEST = 't.test',
  TITLE = '46 gene sig')

```
### 46 gene signature for Event Free Survival ( split by quartile )
```{r}

# get quartiles
q <- quantile(nbl$dinu_efstime, na.rm = T)

# Define comparison
nbl$comp <- ifelse(nbl$dinu_efstime >= q[3], '3', '2')
nbl$comp <- ifelse(nbl$dinu_efstime >= q[4], '4', nbl$comp)
nbl$comp <- ifelse(nbl$dinu_efstime < q[2], '1', nbl$comp)
dat <- nbl %>% dplyr::filter(!is.na(dinu_efstime))


# plot PCA and boxplots
make_plots(
  DF = dat,
  UPGENES = sig46_up, 
  DOWNGENES = sig46_dn, 
  COMP = 'comp', 
  TIMEPOINT = 'all', 
  STAT_TEST = 'anova',
  TITLE = '46 gene sig')

```
### 46 gene signature for Overall Survival ( split by mean)
```{r}

# Quantiles of EFS time
q <- quantile(nbl$dinu_stime, na.rm = T)

# Define comparison
nbl$comp <- ifelse(nbl$dinu_stime >= q[3], 'Long OS', 'Short OS')
dat <- nbl %>% dplyr::filter(!is.na(dinu_stime))


# plot PCA and boxplots
make_plots(
  DF = dat,
  UPGENES = sig46_up, 
  DOWNGENES = sig46_dn, 
  COMP = 'comp', 
  TIMEPOINT = 'all', 
  STAT_TEST = 't.test',
  TITLE = '46 gene sig')

```
### 46 gene signature for Overall Survival ( split by quartile )
```{r}

# get quartiles
q <- quantile(nbl$dinu_stime, na.rm = T)

# Define comparison
nbl$comp <- ifelse(nbl$dinu_stime >= q[3], '3', '2')
nbl$comp <- ifelse(nbl$dinu_stime >= q[4], '4', nbl$comp)
nbl$comp <- ifelse(nbl$dinu_stime < q[2], '1', nbl$comp)
dat <- nbl %>% dplyr::filter(!is.na(dinu_stime))


# plot PCA and boxplots
make_plots(
  DF = dat,
  UPGENES = sig46_up, 
  DOWNGENES = sig46_dn, 
  COMP = 'comp', 
  TIMEPOINT = 'all', 
  STAT_TEST = 'anova',
  TITLE = '46 gene sig')

```

## 96 gene signature

### Load gene signature
```{r}
sig96 <- fread('/mctp/share/users/rebernrj/projects/ferroptosisGeneSig/out/gene_signatures/erastin_elife_correlation_gse104462_96genes.txt',
                data.table = F)

# Upregulated
sig96_up <- sig96 %>% 
  dplyr::filter(GSE104462_LFC > 0) %>% 
  dplyr::select(V1) %>% unlist()

# Downregulated
sig96_dn <- sig96 %>% 
  dplyr::filter(GSE104462_LFC < 0) %>% 
  dplyr::select(V1) %>% unlist()


```
### 96 gene signature for LR/IR/HR
```{r}

# Define comparison
nbl$comp <- nbl$group

# plot PCA and boxplots
make_plots(
  DF = nbl,
  UPGENES = sig96_up, 
  DOWNGENES = sig96_dn, 
  COMP = 'comp', 
  TIMEPOINT = 'dxprim', 
  STAT_TEST = 'anova',
  TITLE = '96 gene sig')

```
### 96 gene signature for Dinutuxumab response among HR
```{r}

# Define comparison
nbl$comp <- nbl$dinuresp_old
dat <- nbl %>% dplyr::filter(!comp %in% c('nothr', 'pending'))

# plot PCA and boxplots
make_plots(
  DF = dat,
  UPGENES = sig96_up, 
  DOWNGENES = sig96_dn, 
  COMP = 'comp', 
  TIMEPOINT = 'dxprim', 
  STAT_TEST = 'anova',
  TITLE = '96 gene sig')

```
### 96 gene signature for Dinutuxumab response pre/post Dinituximab
```{r}

# Define comparison
nbl$comp <- nbl$dinu_treat_prepost
dat <- nbl %>% dplyr::filter(comp %in% c('postDIN', 'preDIN'))

# plot PCA and boxplots
make_plots(
  DF = dat,
  UPGENES = sig96_up, 
  DOWNGENES = sig96_dn, 
  COMP = 'comp', 
  TIMEPOINT = 'all', 
  STAT_TEST = 't.test',
  TITLE = '96 gene sig')

```
### 96 gene signature for Event Free Survival ( split by mean)
```{r}

# Quantiles of EFS time
q <- quantile(nbl$dinu_efstime, na.rm = T)

# Define comparison
nbl$comp <- ifelse(nbl$dinu_efstime >= q[3], 'Long EFS', 'Short EFS')
dat <- nbl %>% dplyr::filter(!is.na(dinu_efstime))


# plot PCA and boxplots
make_plots(
  DF = dat,
  UPGENES = sig96_up, 
  DOWNGENES = sig96_dn, 
  COMP = 'comp', 
  TIMEPOINT = 'all', 
  STAT_TEST = 't.test',
  TITLE = '96 gene sig')

```
### 96 gene signature for Event Free Survival ( split by quartile )
```{r}

# get quartiles
q <- quantile(nbl$dinu_efstime, na.rm = T)

# Define comparison
nbl$comp <- ifelse(nbl$dinu_efstime >= q[3], '3', '2')
nbl$comp <- ifelse(nbl$dinu_efstime >= q[4], '4', nbl$comp)
nbl$comp <- ifelse(nbl$dinu_efstime < q[2], '1', nbl$comp)
dat <- nbl %>% dplyr::filter(!is.na(dinu_efstime))


# plot PCA and boxplots
make_plots(
  DF = dat,
  UPGENES = sig96_up, 
  DOWNGENES = sig96_dn, 
  COMP = 'comp', 
  TIMEPOINT = 'all', 
  STAT_TEST = 'anova',
  TITLE = '96 gene sig')

```
### 96 gene signature for Overall Survival ( split by mean)
```{r}

# Quantiles of EFS time
q <- quantile(nbl$dinu_stime, na.rm = T)

# Define comparison
nbl$comp <- ifelse(nbl$dinu_stime >= q[3], 'Long OS', 'Short OS')
dat <- nbl %>% dplyr::filter(!is.na(dinu_stime))


# plot PCA and boxplots
make_plots(
  DF = dat,
  UPGENES = sig96_up, 
  DOWNGENES = sig96_dn, 
  COMP = 'comp', 
  TIMEPOINT = 'all', 
  STAT_TEST = 't.test',
  TITLE = '96 gene sig')

```
### 96 gene signature for Overall Survival ( split by quartile )
```{r}

# get quartiles
q <- quantile(nbl$dinu_stime, na.rm = T)

# Define comparison
nbl$comp <- ifelse(nbl$dinu_stime >= q[3], '3', '2')
nbl$comp <- ifelse(nbl$dinu_stime >= q[4], '4', nbl$comp)
nbl$comp <- ifelse(nbl$dinu_stime < q[2], '1', nbl$comp)
dat <- nbl %>% dplyr::filter(!is.na(dinu_stime))


# plot PCA and boxplots
make_plots(
  DF = dat,
  UPGENES = sig96_up, 
  DOWNGENES = sig96_dn, 
  COMP = 'comp', 
  TIMEPOINT = 'all', 
  STAT_TEST = 'anova',
  TITLE = '96 gene sig')

```



